---
title: "Diversity-Productivity Relationships in Muskegon Lake"
author: "Marian L. Schmidt"
date: "January 2017"
output:
  html_document:
    code_folding: show
    highlight: haddock
    keep_md: yes
    theme: united
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
      toc_depth: 3
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(eval = TRUE, 
                      echo = FALSE, 
                      cache = TRUE,
                      include = TRUE,
                      warning = FALSE,
                      collapse = FALSE,
                      dependson = NULL,
                      message = FALSE,
                      engine = "R", # Chunks will always have R code, unless noted
                      error = TRUE,
                      fig.path="Figures/cached/",  # Set the figure options
                      fig.align = "center", 
                      fig.width = 7,
                      fig.height = 7)
```


```{r load-libraries}
library(ggplot2)
library(devtools)
library(phyloseq)
library(vegan)
library(tidyr)
library(dplyr)
library(DT)
library(cowplot)
library(wesanderson) # for pretty colors
library(viridis)
library(data.table)
library(stringr)
library(ape)
library(iNEXT) # iNEXT package for alpha diversity measurements 
source("Muskegon_functions.R")
source("set_colors.R")
```

## Load Mothur OTU Data 
```{r load-data-files, echo = TRUE}
# Loads a phyloseq object named otu_merged_musk_pruned)
load("../data/otu_merged_musk_pruned.RData")
# The name of the phyloseq object is: 
otu_merged_musk_pruned 
```


# How do the *summed* sample sequencing read counts vary before modification?
```{r seq-read-count, fig.height=4, fig.width=10, echo = TRUE}
# Check the sequencing depth of each sample 
sums_otu <- data.frame(rowSums(otu_table(otu_merged_musk_pruned)))
colnames(sums_otu) <- "Sample_TotalSeqs"
sums_otu$names <- row.names(sums_otu)
sums_otu <- arrange(sums_otu, Sample_TotalSeqs) 
sums_otu <- make_metadata_norep(sums_otu)

##  PLOT BASED ON 
plot_sample_sums(dataframe = sums_otu, x_total_seqs = "Sample_TotalSeqs", fill_variable = "project")
## YEAR
plot_sample_sums(dataframe = sums_otu, x_total_seqs = "Sample_TotalSeqs", fill_variable = "year")
## FRACTION
plot_sample_sums(dataframe = sums_otu, x_total_seqs = "Sample_TotalSeqs", fill_variable = "fraction")

####  Create a plot of the number of sequences per sample
total_sums <- ggplot(sums_otu, aes(x=reorder(names, Sample_TotalSeqs), y = Sample_TotalSeqs)) + 
  ylab("# of Seqs per Sample") +
  geom_bar(stat = "identity", colour="black",fill="cornflowerblue")  + xlab("Sample Name") + 
  ggtitle("All Samples: Sequencing Depth") + 
  theme(axis.text.x = element_blank())  

sums_lessthan10000 <- ggplot(filter(sums_otu, Sample_TotalSeqs < 10000),
                                    aes(x=reorder(names, Sample_TotalSeqs), y = Sample_TotalSeqs)) + 
  ylab("# of Seqs per Sample") +
  geom_bar(stat = "identity", colour="black",fill="cornflowerblue")  + xlab("Sample Name") + 
  ggtitle("Samples with less \n than 10,000 reads") + 
  theme(axis.text.x = element_blank())  

# DRAW THE TWO PLOTS 
ggdraw() +
  draw_plot(total_sums, 0, 0, 0.7, 1) +
  draw_plot(sums_lessthan10000, 0.7, 0, 0.3, 1) + # 1st = where to start drawing, 3rd = width, 4th = height
  draw_plot_label(c("A", "B"), 
                  c(0, 0.7), # Where along the x-axis would you like the labels?
                  c(1, 1),  # Put the label at the top of the plotting space (y-axis)
                  size = 15) # Size of label

## Add total sequences to metadata frame 
metdf <- sample_data(otu_merged_musk_pruned)
sums_otu$norep_filter_name <- sums_otu$names
sums_otu2 <- select(sums_otu, norep_filter_name, Sample_TotalSeqs)
metdf_num2 <- left_join(metdf, sums_otu2, by = "norep_filter_name")
row.names(metdf_num2) <- metdf_num2$norep_filter_name
# Rename the sample data 
sample_data(otu_merged_musk_pruned) <- metdf_num2
```


# How do the *scaled* (for beta diversity) sample sequencing read counts vary?
```{r scaled_reads_seq_depth, fig.align='center', fig.height=4, fig.width=10, echo = TRUE}
## ALL SAMPLES 
otu_merged_musk_pruned_noMOTHJ715 <- subset_samples(otu_merged_musk_pruned, norep_filter_name !="MOTHJ715")
min(sample_sums(otu_merged_musk_pruned_noMOTHJ715)) # Scaling value 
scale_otu_merged_musk_pruned <- scale_reads(otu_merged_musk_pruned_noMOTHJ715, round = "matround")
scale_otu_merged_musk_pruned # ALL Samples phyloseq object

### Water Samples 
otu_merged_musk_pruned_nosed <- subset_samples(otu_merged_musk_pruned, 
                                               norep_filter_name !="MOTHJ715" & fraction != "Sediment")
min(sample_sums(otu_merged_musk_pruned_nosed)) # Scaling value 
scale_otu_merged_musk_nosed <- scale_reads(otu_merged_musk_pruned_nosed, round = "matround")
scale_otu_merged_musk_nosed # Water samples only

### Particle Samples 
otu_merged_musk_pruned_particle <- subset_samples(otu_merged_musk_pruned, 
                                               norep_filter_name !="MOTHJ715" & 
                                                 fraction %in% c("WholePart", "Particle"))
min(sample_sums(otu_merged_musk_pruned_particle)) # Scaling value 
scale_otu_merged_musk_particle <- scale_reads(otu_merged_musk_pruned_particle, round = "matround")
scale_otu_merged_musk_particle # Particle and Whole particle samples only (both surface and bottom) 

# To only do analysis on wholeparticle samples 
otu_merged_musk_pruned_wholepart_top <- subset_samples(otu_merged_musk_pruned, 
                                               norep_filter_name !="MOTHJ715" & 
                                                 fraction == "WholePart" &
                                                 limnion == "Top")
min(sample_sums(otu_merged_musk_pruned_wholepart_top)) # Scaling value 
scale_otu_merged_musk_wholepart_top <- scale_reads(otu_merged_musk_pruned_wholepart_top, round = "matround")
scale_otu_merged_musk_wholepart_top # Wholeparticle samples only (no prefilter; from the surface only)

### Free-Living Samples 
otu_merged_musk_pruned_free <- subset_samples(otu_merged_musk_pruned, 
                                               norep_filter_name !="MOTHJ715" & 
                                                fraction %in% c("WholeFree", "Free"))
min(sample_sums(otu_merged_musk_pruned_free)) # Scaling value 
scale_otu_merged_musk_free <- scale_reads(otu_merged_musk_pruned_free, round = "matround")
scale_otu_merged_musk_free # All Free living samples (Free and wholeFree; both surface and bottom)

# To only do analysis on WholeFree samples 
otu_merged_musk_pruned_wholefree_top <- subset_samples(otu_merged_musk_pruned, 
                                               norep_filter_name !="MOTHJ715" & 
                                                 fraction == "WholeFree" &
                                                 limnion == "Top")
min(sample_sums(otu_merged_musk_pruned_wholefree_top)) # Scaling value 
scale_otu_merged_musk_wholefree_top <- scale_reads(otu_merged_musk_pruned_wholefree_top, round = "matround")
scale_otu_merged_musk_wholefree_top # Only WholeFree samples (from the surface)


# Check the sequencing depth of each sample 
scaled_sums_otu <- data.frame(rowSums(otu_table(scale_otu_merged_musk_pruned)))
colnames(scaled_sums_otu) <- "Sample_TotalSeqs"
scaled_sums_otu$names <- row.names(scaled_sums_otu)
scaled_sums_otu <- arrange(scaled_sums_otu, Sample_TotalSeqs) %>%
  make_metadata_norep()

##  Plot based on all samples 
plot_sample_sums(dataframe = scaled_sums_otu, x_total_seqs = "Sample_TotalSeqs", fill_variable = "project")
##  Plot based on depth of samples 
plot_sample_sums(dataframe = scaled_sums_otu, x_total_seqs = "Sample_TotalSeqs", fill_variable = "limnion")
```

  
```{r create_metadata_dataframes}
# Create the metadata information 
meta_data <- data.frame(sample_data(otu_merged_musk_pruned)) 

# Simpson's Evenness
meta_data <- meta_data %>%
  mutate(simps_even = D2/D0)

### PREPARE DATA FRAMES FOR PHENOFLOW ALPHA DIVERSITY ANALYSIS
free_meta_data <- filter(meta_data, fraction %in% c("Free", "WholeFree") & norep_filter_name != "MOTHJ715" & limnion == "Top")
part_meta_data <- filter(meta_data, fraction %in% c("Particle", "WholePart") & norep_filter_name != "MOTHJ715" & limnion == "Top")
nosed_meta_data <- filter(meta_data, fraction != "Sediment" & limnion == "Top")

# Only "True Free Living and Particle" From 20um prefiltered samples 
free_only <- filter(free_meta_data, fraction == "Free")
part_only <- filter(part_meta_data, fraction == "Particle")

## 2015 specific samples that have NOT been prefiltered (whole water)
wholefree_only <- filter(free_meta_data, fraction == "WholeFree")
wholepart_only <- filter(part_meta_data, fraction == "WholePart")

####################### FOR FRACTION DIVERSITY PLOTS 
# How do the diversities compare with each other in terms of their scales?
fraction_divs <- nosed_meta_data %>%
  filter(limnion == "Top") %>%
  dplyr::select(1:7,norep_filter_name, 60:69) %>%
  gather(div_metric, div_value, D0, D0_chao, D1, D2) %>%
  select(-norep_water_name)

# Set the factor levels to be the order we'd like
fraction_divs$fraction <- factor(fraction_divs$fraction,  levels = c("WholePart", "Particle", "WholeFree", "Free"))
fraction_divs$lakesite <- factor(fraction_divs$lakesite,  levels = c("MOT", "MDP", "MBR", "MIN"))
```


#  How linked are the diversities of the fractions?
## Free Living Sample Diversity Comparison  
```{r free_vs_wholefree_diversity, fig.height=8, fig.width=10, echo = TRUE}
# Plot diversity based only on fraction 
ggplot(filter(fraction_divs, fraction %in% c("Free","WholeFree")),
       aes(x = fraction, y = div_value)) + 
  geom_jitter(aes(color = fraction), size = 3, width = 0.2) + 
  geom_boxplot(aes(fill = fraction), alpha =0.5) +
  scale_color_manual(values = fraction_colors) + 
  scale_fill_manual(values = fraction_colors) + 
  facet_wrap(~div_metric, scales = "free_y") +
  theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1),
        legend.position = c(0.07, 0.93), 
        strip.background = element_rect(fill = NA), 
        strip.text.x = element_text(face = "bold"))

# Plot diversity based on lakesite lumped together
ggplot(filter(fraction_divs, fraction %in% c("Free","WholeFree")),
       aes(x = lakesite, y = div_value)) + 
  geom_jitter(aes(color = lakesite), size = 3, width = 0.2) + 
  geom_boxplot(aes(fill = lakesite), alpha =0.5) +
  scale_color_manual(values = lakesite_colors) + 
  scale_fill_manual(values = lakesite_colors) + 
  facet_wrap(~div_metric, scales = "free_y") +
  theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1),
        legend.position = c(0.07, 0.93), 
        strip.background = element_rect(fill = NA), 
        strip.text.x = element_text(face = "bold"))


# Value of Diversity estimate on y-axis and site vs fraction on x-axis
ggplot(filter(fraction_divs, fraction %in% c("Free","WholeFree")),
       aes(x = lakesite, y = div_value)) + 
  geom_jitter(aes(color = fraction), size = 3, position = position_dodge(width = 0.8)) + 
  geom_boxplot(aes(fill = fraction), alpha =0.5) +
  scale_color_manual(values = fraction_colors) + 
  scale_fill_manual(values = fraction_colors) + 
  facet_wrap(~div_metric, scales = "free_y") +
  theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1),
        legend.position = c(0.07, 0.93), 
        strip.background = element_rect(fill = NA), 
        strip.text.x = element_text(face = "bold"))

###  ARE THERE DIFFERENCES BETWEEN FRACTION AND LAKESITE WITHIN THE FREE LIVING?
free_fraction_divs <- filter(fraction_divs, fraction %in% c("Free","WholeFree"))
#### Significant difference between D0 lakesite and fraction?
summary(aov(div_value ~ fraction*lakesite,  
            data = filter(free_fraction_divs, div_metric == "D0")))
#### Significant difference between D0_chao lakesite and fraction?
summary(aov(div_value ~ fraction*lakesite,  
            data = filter(free_fraction_divs, div_metric == "D0_chao")))
#### Significant difference between D1 lakesite and fraction?
summary(aov(div_value ~ fraction*lakesite,  
            data = filter(free_fraction_divs, div_metric == "D1")))
#### Significant difference between D2 lakesite and fraction?
summary(aov(div_value ~ fraction*lakesite,  
            data = filter(free_fraction_divs, div_metric == "D2")))


# Prepare data frames for plotting diversity of wholefree vs free 
free_samps_alpha <- filter(free_meta_data, fraction == "Free" & year == "2015") %>% 
  dplyr::select(norep_filter_name, fraction, D0, D0_chao, D1, D2, simps_even) %>%
  # Make a new column called "div_metric" that includes the hill diversity metric
  gather("div_metric", "value", D0, D0_chao, D1, D2, simps_even) %>%
  # Combine the diversity metric and fraction into one column called frac_div_metric
  unite("frac_div_metric", fraction, div_metric, sep = "_") %>%
  # Spread frac_div_metric into 4 columns for each hill div metric
  spread(frac_div_metric, value) %>%
  mutate(norep_water_name = paste(substr(norep_filter_name,1,4), substr(norep_filter_name, 6, 8), sep = "")) %>%
  dplyr::select(-norep_filter_name)

# Prepare data frames for plotting diversity of wholefree vs free 
wholefree_samps_alpha <- filter(free_meta_data, fraction == "WholeFree" & year == "2015") %>% 
  dplyr::select(norep_filter_name, fraction, D0, D0_chao, D1, D2, simps_even) %>%
  # Make a new column called "div_metric" that includes the hill diversity metric
  gather(div_metric, value, D0, D1, D2, D0_chao, simps_even) %>%
  # Combine the diversity metric and fraction into one column called frac_div_metric
  unite(frac_div_metric, fraction, div_metric, sep = "_") %>%
  # Spread frac_div_metric into 4 columns for each hill div metric
  spread(frac_div_metric, value) %>%
  mutate(norep_water_name = paste(substr(norep_filter_name,1,4), substr(norep_filter_name, 6, 8), sep = "")) %>%
  dplyr::select(-norep_filter_name)

# join the data frames together 
FL_alpha <- left_join(free_samps_alpha, wholefree_samps_alpha, by = "norep_water_name")


# Is there a linear relationship between free and whole free D0 diversity?
lm_FL_D0 <- lm(Free_D0 ~ WholeFree_D0, data = FL_alpha)
# Plot linear relationship between free and whole free D0 diversity
free_fraction_D0 <- ggplot(FL_alpha, aes(x = WholeFree_D0, y = Free_D0)) + 
  geom_point(size = 3) + geom_smooth(method = "lm") +
  xlim(300, 2000) + ylim(300, 2000) + 
  geom_abline(slope = 1, intercept = 0) +
  annotate("text", x = 1000, y=500, color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_FL_D0)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_FL_D0)$coefficients[,4][2]), digits = 8)))


# Is there a linear relationship between free and whole free D0_chao diversity?
lm_FL_simps_D0_chao <- lm(Free_D0_chao ~ WholeFree_D0_chao, data = FL_alpha)
# Plot linear relationship between free and whole free D0 diversity
free_fraction_D0_chao <- ggplot(FL_alpha, aes(x = WholeFree_D0_chao, y = Free_D0_chao)) + 
  geom_point(size = 3) + geom_smooth(method = "lm") +
  xlim(300, 3600) + ylim(300, 3600) +
  geom_abline(slope = 1, intercept = 0) +
  annotate("text", x = 3000, y=500, color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_FL_simps_D0_chao)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_FL_simps_D0_chao)$coefficients[,4][2]), digits = 5)))


# Is there a linear relationship between free and whole free D1 diversity?
lm_FL_D1 <- lm(Free_D1 ~ WholeFree_D1, data = FL_alpha)
# Plot linear relationship between free and whole free D1 diversity
free_fraction_D1 <- ggplot(FL_alpha, aes(x = WholeFree_D1, y = Free_D1)) + 
  geom_point(size = 3) + geom_smooth(method = "lm") +
  xlim(0, 175) + ylim(0,175) + 
  geom_abline(slope = 1, intercept = 0) +
  annotate("text", x = 100, y=25, color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_FL_D1)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_FL_D1)$coefficients[,4][2]), digits = 12)))

# Is there a linear relationship between free and whole free D2 diversity?
lm_FL_D2 <- lm(Free_D2 ~ WholeFree_D2, data = FL_alpha)
# Plot linear relationship between free and whole free D2 diversity
free_fraction_D2 <- ggplot(FL_alpha, aes(x = WholeFree_D2, y = Free_D2)) + 
  geom_point(size = 3) + geom_smooth(method = "lm") +
  xlim(10, 55) + ylim(10, 55) +
  geom_abline(slope = 1, intercept = 0) +
  annotate("text", x = 35, y=17, color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_FL_D2)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_FL_D2)$coefficients[,4][2]), digits = 8)))
  
free_vs_freepart_div_plots <- plot_grid(free_fraction_D0, free_fraction_D0_chao, free_fraction_D1, free_fraction_D2, 
          labels = c("A", "B", "C", "D"), ncol = 2, nrow = 2)

free_vs_freepart_div_plots
#ggsave("../Figures/free_vs_wholefree_div_plots.png", plot = free_vs_freepart_div_plots, dpi = 600, width = 10, height = 8)

```

##  Particle-Associated Diversity Comparison 
```{r part_vs_wholepart_diversity, fig.height=8, fig.width=10, echo = TRUE}
# Plot diversity based only on fraction 
ggplot(filter(fraction_divs, fraction %in% c("Particle","WholePart")),
       aes(x = fraction, y = div_value)) + 
  geom_jitter(aes(color = fraction), size = 3, width = 0.2) + 
  geom_boxplot(aes(fill = fraction), alpha =0.5) +
  scale_color_manual(values = fraction_colors) + 
  scale_fill_manual(values = fraction_colors) + 
  facet_wrap(~div_metric, scales = "free_y") +
  theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1),
        legend.position = c(0.07, 0.93), 
        strip.background = element_rect(fill = NA), 
        strip.text.x = element_text(face = "bold"))

# Plot based on lakesite lumped together
ggplot(filter(fraction_divs, fraction %in% c("Particle","WholePart")),
       aes(x = lakesite, y = div_value)) + 
  geom_jitter(aes(color = lakesite), size = 3, width = 0.2) + 
  geom_boxplot(aes(fill = lakesite), alpha =0.5) +
  scale_color_manual(values = lakesite_colors) + 
  scale_fill_manual(values = lakesite_colors) + 
  facet_wrap(~div_metric, scales = "free_y") +
  theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1),
        legend.position = c(0.07, 0.93), 
        strip.background = element_rect(fill = NA), 
        strip.text.x = element_text(face = "bold"))


# Value of Diversity estimate on y-axis and site vs fraction on x-axis
ggplot(filter(fraction_divs, fraction %in% c("Particle","WholePart")),
       aes(x = lakesite, y = div_value)) + 
  geom_jitter(aes(color = fraction), size = 3, position = position_dodge(width = 0.8)) + 
  geom_boxplot(aes(fill = fraction), alpha =0.5) +
  scale_color_manual(values = fraction_colors) + 
  scale_fill_manual(values = fraction_colors) + 
  facet_wrap(~div_metric, scales = "free_y") +
  theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1),
        legend.position = c(0.07, 0.93), 
        strip.background = element_rect(fill = NA), 
        strip.text.x = element_text(face = "bold"))


###  ARE THERE DIFFERENCES BETWEEN FRACTION AND LAKESITE WITHIN PARTICLE ASSOCIATION?
part_fraction_divs <- filter(fraction_divs, fraction %in% c("Particle","WholePart"))

#### Significant difference between D0 lakesite and fraction?
summary(aov(div_value ~ fraction*lakesite,  
            data = filter(part_fraction_divs, div_metric == "D0")))
#### Significant difference between D0_chao lakesite and fraction?
summary(aov(div_value ~ fraction*lakesite,  
            data = filter(part_fraction_divs, div_metric == "D0_chao")))
#### Significant difference between D1 lakesite and fraction?
summary(aov(div_value ~ fraction*lakesite,  
            data = filter(part_fraction_divs, div_metric == "D1")))
#### Significant difference between D2 lakesite and fraction?
summary(aov(div_value ~ fraction*lakesite,  
            data = filter(part_fraction_divs, div_metric == "D2")))


# Prepare data frames for plotting diversity of wholepart vs particle 
part_samps_alpha <- filter(part_meta_data, fraction == "Particle" & year == "2015") %>% # & norep_filter_name != "MBRHJ515"
  select(norep_filter_name, fraction, D0, D0_chao, D1, D2, simps_even) %>%
  gather("div_metric", "value", D0, D0_chao, D1, D2, simps_even) %>%
  unite("frac_div_metric", fraction, div_metric, sep = "_") %>%
  spread(frac_div_metric, value) %>%
  mutate(norep_water_name = paste(substr(norep_filter_name,1,4), substr(norep_filter_name, 6, 8), sep = "")) %>%
  select(-norep_filter_name)

# Prepare data frames for plotting diversity of wholepart vs particle 
wholepart_samps_alpha <- filter(part_meta_data, fraction == "WholePart" & year == "2015") %>% # & norep_filter_name != "MBRHJ515"
  select(norep_filter_name, fraction, D0, D0_chao, D1, D2, simps_even) %>%
  gather(div_metric, value, D0, D0_chao, D1, D2, simps_even) %>%
  unite(frac_div_metric, fraction, div_metric, sep = "_") %>%
  spread(frac_div_metric, value) %>%
  mutate(norep_water_name = paste(substr(norep_filter_name,1,4), substr(norep_filter_name, 6, 8), sep = "")) %>%
  select(-norep_filter_name)

# Combine the data frames into one
PA_alpha <- left_join(part_samps_alpha, wholepart_samps_alpha, by = "norep_water_name")

# Is there a linear relationship between particle and WholePart D0 diversity?
lm_PA_D0 <- lm(Particle_D0 ~ WholePart_D0, data = PA_alpha)
# Plot the linear relationship between particle and WholePart D0 diversity
PA_fraction_D0 <- ggplot(PA_alpha, aes(x = WholePart_D0, y = Particle_D0)) + 
  geom_point(size = 3) + geom_smooth(method = "lm", se = FALSE) +
  xlim(0, 2200) + ylim(0, 2200) + 
  geom_abline(slope = 1, intercept = 0) +
  annotate("text", x = 1500, y=300, color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_PA_D0)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_PA_D0)$coefficients[,4][2]), digits = 4)))


# Is there a linear relationship between particle and WholePart D0_chao diversity?
lm_PA_D0_chao <- lm(Particle_D0_chao ~ WholePart_D0_chao, data = PA_alpha)
# Plot the linear relationship between particle and WholePart D0_chao diversity
PA_fraction_D0_chao <- ggplot(PA_alpha, aes(x = WholePart_D0_chao, y = Particle_D0_chao)) + 
  geom_point(size = 3) + geom_smooth(method = "lm", se = FALSE) +
  xlim(0, 2200) + ylim(0, 2200) + 
  geom_abline(slope = 1, intercept = 0) +
  annotate("text", x = 1500, y=300, color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_PA_D0_chao)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_PA_D0_chao)$coefficients[,4][2]), digits = 4)))


# Is there a linear relationship between particle and WholePart D1 diversity?
lm_PA_D1 <- lm(Particle_D1 ~ WholePart_D1, data = PA_alpha)
# Plot the linear relationship between particle and WholePart D1 diversity
PA_fraction_D1 <- ggplot(PA_alpha, aes(x = WholePart_D1, y = Particle_D1)) + 
  geom_point(size = 3) + geom_smooth(method = "lm") + 
  xlim(0, 620) + ylim(0,620) + 
  geom_abline(slope = 1, intercept = 0) +
  annotate("text", x = 150, y=500, color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_PA_D1)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_PA_D1)$coefficients[,4][2]), digits = 5)))

# Is there a linear relationship between particle and WholePart D2 diversity?
lm_PA_D2 <- lm(Particle_D2 ~ WholePart_D2, data = PA_alpha)
# Plot the linear relationship between particle and WholePart D2 diversity
PA_fraction_D2 <- ggplot(PA_alpha, aes(x = WholePart_D2, y = Particle_D2)) + 
  geom_point(size = 3) + geom_smooth(method = "lm") +
  xlim(10, 250) + ylim(10, 250) +
  geom_abline(slope = 1, intercept = 0) +
  annotate("text", x = 50, y=175, color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_PA_D2)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_PA_D2)$coefficients[,4][2]), digits = 5)))

lm_PA_simps_even <- lm(Particle_simps_even ~ WholePart_simps_even, data = PA_alpha)

PA_fraction_simps_even <- ggplot(PA_alpha, aes(x = WholePart_simps_even, y = Particle_simps_even)) + 
  geom_point(size = 3) + geom_smooth(method = "lm", se = FALSE) +
  xlim(0.01, 0.2) + ylim(0.01, 0.2) +
  geom_abline(slope = 1, intercept = 0) +
  annotate("text", x = 0.06, y=0.12, color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_PA_simps_even)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_PA_simps_even)$coefficients[,4][2]), digits = 5)))
  
part_vs_wholepart_div_plots <- plot_grid(PA_fraction_D0, PA_fraction_D0_chao, PA_fraction_D1, PA_fraction_D2, 
          labels = c("A", "B", "C", "D"), ncol = 2, nrow = 2)

part_vs_wholepart_div_plots
#ggsave("../Figures/part_vs_wholepart_div_plots.png", plot = part_vs_wholepart_div_plots, dpi = 600, width = 10, height = 8)
```


# Is there a significant relationship between sequencing depth and diversity?
```{r div_vs_seqdepth, fig.width=10, fig.height = 4, echo = TRUE}
## Is there a significant relationship between sequencing depth and D0?
summary(lm(D0 ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "Free"))) # p = 0.002 but R2 = -0.09 
#summary(lm(D0 ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "WholeFree"))) # NS
summary(lm(D0 ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "Particle"))) # Significant!
#summary(lm(D0 ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "WholePart"))) # NS

# Plot Sequencing depth vs D0
ggplot(nosed_meta_data, aes(x = Sample_TotalSeqs, y = D0)) + 
  geom_point(size = 3) + 
  facet_grid(.~fraction, scale = "free_x") + 
  geom_smooth(method = "lm", data = filter(nosed_meta_data, fraction %in% c("Free", "Particle"))) + 
  ggtitle("D0: Species Richness") + 
  theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1),
        legend.position = c(0.1, 0.9), 
        strip.background = element_rect(fill = NA), 
        strip.text.x = element_text(face = "bold"))

## Is there a significant relationship between sequencing depth and D0_chao?
summary(lm(D0_chao ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "Free"))) # pval = 0.006, R2 = 0.27  
#summary(lm(D0_chao ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "WholeFree"))) # NS
#summary(lm(D0_chao ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "Particle"))) # NS
#summary(lm(D0_chao ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "WholePart"))) #  NS

# Plot the relationship between D0_chao and sequencing depth 
ggplot(nosed_meta_data, aes(x = Sample_TotalSeqs, y = D0_chao)) + 
  geom_point(size = 3) + 
  facet_grid(.~fraction, scale = "free_x") + 
  geom_smooth(method = "lm", data = filter(nosed_meta_data, fraction == "Free")) + 
  ggtitle("D0: Chao 1") + 
  theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1),
        legend.position = c(0.1, 0.9), 
        strip.background = element_rect(fill = NA), 
        strip.text.x = element_text(face = "bold"))

## Is there a significant relationship between sequencing depth and D1?
#summary(lm(D1 ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "Free"))) # NS
#summary(lm(D1 ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "WholeFree"))) # NS 
#summary(lm(D1 ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "Particle"))) # NS
#summary(lm(D1 ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "WholePart"))) # NS

# Plot the relationship between D1 and sequencing depth 
ggplot(nosed_meta_data, aes(x = Sample_TotalSeqs, y = D1)) + 
  geom_point(size = 3) + 
  facet_grid(.~fraction, scale = "free_x") + 
  ggtitle("D1: Exponential of Shannon Entropy") + 
  theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1),
        legend.position = c(0.1, 0.9), 
        strip.background = element_rect(fill = NA), 
        strip.text.x = element_text(face = "bold"))

## Is there a significant relationship between sequencing depth and D2?
#summary(lm(D2 ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "Free"))) # NS
#summary(lm(D2 ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "WholeFree"))) # NS 
#summary(lm(D2 ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "Particle"))) # NS
#summary(lm(D2 ~ Sample_TotalSeqs,data = filter(nosed_meta_data, fraction == "WholePart"))) # NS

# Plot the relationship between D2 and sequencing depth 
ggplot(nosed_meta_data, aes(x = Sample_TotalSeqs, y = D2)) + 
  geom_point(size = 3) + 
  facet_grid(.~fraction, scale = "free_x") + 
  ggtitle("D2: Inverse Simpson") + 
  theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1),
        legend.position = c(0.1, 0.9), 
        strip.background = element_rect(fill = NA), 
        strip.text.x = element_text(face = "bold"))

```

# Does DNA extraction concentration influcence diversity?
```{r dnaconc_vs_diversity, fig.width=10, fig.height = 4, echo = TRUE}
### Prepare the data frame 
DNA_comparison <- nosed_meta_data %>%
  dplyr::select(norep_filter_name, dnaconcrep1, dnaconcrep2, cells_per_uL, lakesite) %>%
  mutate(dnaconcrep1 = as.numeric(dnaconcrep1),
         dnaconcrep2 = as.numeric(dnaconcrep2),
         mean_dna = (dnaconcrep1+dnaconcrep2)/2)

# Calculate the linear models 
lm_dna_conc <- lm(dnaconcrep1 ~ dnaconcrep2,  data = DNA_comparison)
lm_cells_vs_meanDNA <- lm(mean_dna ~ cells_per_uL,  data = DNA_comparison)

# Plot the trends 
plot_grid(ggplot(DNA_comparison, aes(x = dnaconcrep1, y = dnaconcrep2)) + # First plot 
            geom_point(size = 3) + geom_smooth(method = "lm") +
            annotate("text",  x = 17, y = 3,
              color = "black", fontface = "bold",
               label = paste("R2 =", round(summary(lm_dna_conc)$adj.r.squared, digits = 4), "\n", 
                             "p =", round(unname(summary(lm_dna_conc)$coefficients[,4][2]), digits = 8)))+
            ggtitle("DNA Concentration Between Replicates"), 
          ggplot(DNA_comparison, aes(x = mean_dna, y = cells_per_uL)) + # Second plot 
            geom_point(size = 3) + geom_smooth(method = "lm") +
            annotate("text",  x = 12, y = 4800,
              color = "black", fontface = "bold",
               label = paste("R2 =", round(summary(lm_cells_vs_meanDNA)$adj.r.squared, digits = 4), "\n", 
                             "p =", round(unname(summary(lm_cells_vs_meanDNA)$coefficients[,4][2]), digits = 4)))+
            ggtitle("Mean DNA Concentration vs Cell Numbers"),
          labels = c("A", "B"), ncol = 2)
```



# Is there a relationship between diversity and productivity?

**Note:** The total production data is only for the surface during 2014 and 2015!

## D0 vs Total Production
```{r D0_totalproduction_vs_diversity, fig.width=10, fig.height = 4, echo = TRUE}
#### FREE LIVING SAMPLES VS TOTAL BACTERIAL PRODUCTION 
# Is there a significant relationship between FL D0 and total production?
lm_freeonly_totprod_D0 <- lm(tot_bacprod ~ D0, data = free_only)
# Plot the relationship 
plot_totprod_free_D0 <-  ggplot(free_only, aes(y = tot_bacprod, x = D0)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  geom_smooth(method = "lm", color = "black") + 
  ggtitle("20 um Prefiltered Free-Living Only") + 
  annotate("text",  x = 2250, y = 15, # For D2:  x = 40, y=5, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_freeonly_totprod_D0)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_freeonly_totprod_D0)$coefficients[,4][2]), digits = 4)))+
  theme(legend.position = c(0.8, 0.2), 
        legend.text = element_text(size = 10))


# Individually for 2014 and 2015, the trend is NS 
# summary(lm(tot_bacprod ~ D0, data = filter(free_only, year == "2014"))) # NS
# summary(lm(tot_bacprod ~ D0, data = filter(free_only, year == "2015"))) # NS

# The trend is close to signifincant!
lm_wholefreeonly_totprod_D0 <- lm(tot_bacprod ~ D0, data = wholefree_only)
# Plot the relationship between wholefree and total production 
plot_totprod_wholefree_D0 <- ggplot(wholefree_only, aes(y = tot_bacprod, x = D0)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  geom_smooth(method = "lm", color = "black") + 
  ggtitle("WholeFree Only") + 
  annotate("text",  x = 800, y = 2, # For D2:  x = 40, y=5, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_wholefreeonly_totprod_D0)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_wholefreeonly_totprod_D0)$coefficients[,4][2]), digits = 4)))+
  theme(legend.position = c(0.12, 0.8), 
        legend.text = element_text(size = 10))


## Prefiltered 2015 free only 
free_only_2015_D0 <- filter(free_only, year == "2015")
lm_free2015_only_totprod_D0 <- lm(tot_bacprod ~ D0, data = free_only_2015_D0) 
plot_totprod_freeonly_2015_D0 <- ggplot(free_only_2015_D0, aes(y = tot_bacprod, x = D0)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  #geom_smooth(method = "lm", color = "black") + 
  ggtitle("2015 Prefiltered Free-Living") + 
  annotate("text", x = 1250, y=20, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_free2015_only_totprod_D0)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_free2015_only_totprod_D0)$coefficients[,4][2]), digits = 4))) +
  theme(legend.position = c(0.12, 0.8), 
        legend.text = element_text(size = 10))



#### PARTICLE ASSOCIATED SAMPLES VS TOTAL BACTERIAL PRODUCTION 
# Is there a significant relationship?
lm_partonly_totprod_D0 <- lm(tot_bacprod ~ D0, data = part_only)
# Plot the relationship 
plot_totprod_part_D0 <- ggplot(part_only, aes(y = tot_bacprod, x = D0)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  ggtitle("2014 & 2015 Prefiltered Particle") + 
  annotate("text", x = 1700, y = 75, # For D2:  x = 40, y=5, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_partonly_totprod_D0)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_partonly_totprod_D0)$coefficients[,4][2]), digits = 4))) +
  theme(legend.position = c(0.8, 0.2), 
        legend.text = element_text(size = 10))

# Trend is NS in 2014 & 2015
#summary(lm(tot_bacprod ~ D0, data = filter(part_only, year == "2014"))) # NS
#summary(lm(tot_bacprod ~ D0, data = filter(part_only, year == "2015"))) # NS

### Different particulate fractions 
## Particle (20-3um) fraction only 
part_only_2015_D0 <- filter(part_only, year == "2015")
lm_part2015_only_totprod_D0 <- lm(tot_bacprod ~ D0, data = part_only_2015_D0) 
plot_totprod_partonly_2015_D0 <- ggplot(part_only_2015_D0, aes(y = tot_bacprod, x = D0)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  #geom_smooth(method = "lm", color = "black") + 
  ggtitle("2015 Prefiltered Particle") + 
  annotate("text", x = 1800, y=15, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_part2015_only_totprod_D0)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_part2015_only_totprod_D0)$coefficients[,4][2]), digits = 4))) +
  theme(legend.position = c(0.12, 0.7), 
        legend.text = element_text(size = 10))



# The trend is signifincant!
lm_wholepartonly_totprod_D0 <- lm(tot_bacprod ~ D0, data = wholepart_only)
# Plot the relationship between wholepart and total production 
plot_totprod_wholepart_D0 <- ggplot(wholepart_only, aes(y = tot_bacprod, x = D0)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  geom_smooth(method = "lm", color = "black") + 
  ggtitle("WholePart Only") + 
  annotate("text",  x = 1400, y = 5, # For D2:  x = 40, y=5, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_wholepartonly_totprod_D0)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_wholepartonly_totprod_D0)$coefficients[,4][2]), digits = 4)))+
  theme(legend.position = c(0.12, 0.7), 
        legend.text = element_text(size = 10))

### Plot it all together for the pre-filtered fractions 
plot_D0_totprod_prefilt <- plot_grid(plot_totprod_free_D0, plot_totprod_part_D0,
          labels = c("A", "B"), ncol = 2)
plot_D0_totprod_prefilt

### Plot it all together for both of the free living fractions fractions 
plot_D0_totprod_FL_comparison <- plot_grid(plot_totprod_freeonly_2015_D0, plot_totprod_wholefree_D0,
          labels = c("A", "B"), ncol = 2)
plot_D0_totprod_FL_comparison

### Plot it all together for both of the particle associated fractions fractions 
plot_D0_totprod_PA_comparison <- plot_grid(plot_totprod_partonly_2015_D0, plot_totprod_wholepart_D0,
          labels = c("A", "B"), ncol = 2)
plot_D0_totprod_PA_comparison

```


## D1 vs Total Production
```{r D1_totalproduction_vs_diversity, fig.width=10, fig.height = 4, echo = TRUE}
#### FREE LIVING SAMPLES VS TOTAL BACTERIAL PRODUCTION 
# Is there a significant relationship?
lm_freeonly_totprod_D1 <- lm(tot_bacprod ~ D1, data = free_only)
summary(lm_freeonly_totprod_D1)

lm_freeonly_totprod_D1_MINE1F514 <- lm(tot_bacprod ~ D1, data = filter(free_only, Sample_16S != "MINE1F514"))

# Plot the relationship 
plot_totprod_free_D1 <-  ggplot(free_only, aes(y = tot_bacprod, x = D1)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  geom_smooth(method = "lm", color = "black") + 
  ggtitle("20 um Prefiltered Free-Living Only") + 
  annotate("text",  x = 120, y = 10, # For D2:  x = 40, y=5, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_freeonly_totprod_D1)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_freeonly_totprod_D1)$coefficients[,4][2]), digits = 4)))+
  theme(legend.position = c(0.12, 0.7), 
        legend.text = element_text(size = 10))
  
## Prefiltered 2015 free only 
free_only_2015_D1 <- filter(free_only, year == "2015")
lm_free2015_only_totprod_D1 <- lm(tot_bacprod ~ D1, data = free_only_2015_D1) 
plot_totprod_freeonly_2015_D1 <- ggplot(free_only_2015_D1, aes(y = tot_bacprod, x = D1)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  #geom_smooth(method = "lm", color = "black") + 
  ggtitle("2015 Prefiltered Free-Living") + 
  annotate("text", x = 90, y=20, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_free2015_only_totprod_D1)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_free2015_only_totprod_D1)$coefficients[,4][2]), digits = 4))) +
  theme(legend.position = c(0.12, 0.8), 
        legend.text = element_text(size = 10))


# Trend is only marginally significant (p = 0.08) in 2014 and NS in 2015
summary(lm(tot_bacprod ~ D1, data = filter(free_only, year == "2014")))
#summary(lm(tot_bacprod ~ D1, data = filter(free_only, year == "2015"))) # NS

# The trend is also insignificant for the whole free fraction 
lm_wholefreeonly_totprod_D1 <- lm(tot_bacprod ~ D1, data = wholefree_only)
# Plot the relationship between wholefree and total production 
plot_totprod_wholefree_D1 <- ggplot(wholefree_only, aes(y = tot_bacprod, x = D1)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  ggtitle("WholeFree Only") + 
  annotate("text",  x = 100, y = 5, # For D2:  x = 40, y=5, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_wholefreeonly_totprod_D1)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_wholefreeonly_totprod_D1)$coefficients[,4][2]), digits = 4)))+
  theme(legend.position = c(0.12, 0.8), 
        legend.text = element_text(size = 10))


#### PARTICLE ASSOCIATED SAMPLES VS TOTAL BACTERIAL PRODUCTION 
# Is there a significant relationship?
lm_partonly_totprod_D1 <- lm(tot_bacprod ~ D1, data = part_only)
#summary(lm_partonly_totprod_D1)
# Plot the relationship 
plot_totprod_part_D1 <- ggplot(part_only, aes(y = tot_bacprod, x = D1)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  ggtitle("Particle (3-20 um) Samples Only") + 
  annotate("text", x = 175, y = 15, # For D2:  x = 40, y=5, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_partonly_totprod_D1)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_partonly_totprod_D1)$coefficients[,4][2]), digits = 4))) +
  theme(legend.position = c(0.12, 0.7), 
        legend.text = element_text(size = 10))

# Trend is NS in 2014 & 2015
# summary(lm(tot_bacprod ~ D1, data = filter(part_only, year == "2014"))) # NS
# summary(lm(tot_bacprod ~ D1, data = filter(part_only, year == "2015")))  # NS


### Different particulate fractions 
## Particle (20-3um) fraction only 
part_only_2015_D1 <- filter(part_only, year == "2015")
lm_part2015_only_totprod_D1 <- lm(tot_bacprod ~ D1, data = part_only_2015_D1) 
plot_totprod_partonly_2015_D1 <- ggplot(part_only_2015_D1, aes(y = tot_bacprod, x = D1)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  #geom_smooth(method = "lm", color = "black") + 
  ggtitle("2015 20um Prefiltered Particle") + 
  annotate("text", x = 175, y=15, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_part2015_only_totprod_D1)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_part2015_only_totprod_D1)$coefficients[,4][2]), digits = 4))) +
  theme(legend.position = c(0.12, 0.7), 
        legend.text = element_text(size = 10))

# The trend is close to significant for whole particle fraction 
lm_wholepartonly_totprod_D1 <- lm(tot_bacprod ~ D1, data = wholepart_only)
# Plot the relationship between wholefree and total production 
plot_totprod_wholepart_D1 <- ggplot(wholepart_only, aes(y = tot_bacprod, x = D1)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  geom_smooth(method = "lm", color = "black") + 
  ggtitle("WholePart Only") + 
  annotate("text",  x = 275, y = 10, # For D2:  x = 40, y=5, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_wholepartonly_totprod_D1)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_wholepartonly_totprod_D1)$coefficients[,4][2]), digits = 4)))+
  theme(legend.position = c(0.12, 0.7), 
        legend.text = element_text(size = 10))



### Plot it all together for the pre-filtered fractions 
plot_D1_totprod_prefilt <- plot_grid(plot_totprod_free_D1, plot_totprod_part_D1,
          labels = c("A", "B"), ncol = 2)
plot_D1_totprod_prefilt

### Plot it all together for both of the free living fractions fractions 
plot_D1_totprod_FL_comparison <- plot_grid(plot_totprod_freeonly_2015_D1, plot_totprod_wholefree_D1,
          labels = c("A", "B"), ncol = 2)
plot_D1_totprod_FL_comparison

### Plot it all together for both of the particle associated fractions fractions 
plot_D1_totprod_PA_comparison <- plot_grid(plot_totprod_partonly_2015_D1, plot_totprod_wholepart_D1,
          labels = c("A", "B"), ncol = 2)
plot_D1_totprod_PA_comparison
```

### An important note:
For D1 free living fraction (plot entitled "20 um Prefiltered Free-Living Only") there seems to be an outlier along both axes.  The regression without this point results in an **R2 of `r round(summary(lm_freeonly_totprod_D1_MINE1F514)$adj.r.squared, digits = 4)`, a pvalue of `r round(unname(summary(lm_freeonly_totprod_D1_MINE1F514)$coefficients[,4][2]), digits = 4)`**.



## D2 vs Total Production
```{r D2_totalproduction_vs_diversity,fig.width=10, fig.height = 4, echo = TRUE}
#### FREE LIVING SAMPLES VS TOTAL BACTERIAL PRODUCTION 
# Is there a significant relationship?
lm_freeonly_totprod_D2 <- lm(tot_bacprod ~ D2, data = free_only)
#summary(lm_freeonly_totprod_D2)
# Plot the relationship
plot_totprod_free_D2 <- ggplot(free_only, aes(y = tot_bacprod, x = D2)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  xlim(0, 60) + # FOR D2
  ggtitle("20 um Prefiltered Free-Living Only") + 
  annotate("text", x = 48, y=15, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_freeonly_totprod_D2)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_freeonly_totprod_D2)$coefficients[,4][2]), digits = 4)))+
  theme(legend.position = c(0.12, 0.8), 
        legend.text = element_text(size = 10))


## Prefiltered 2015 free only 
free_only_2015_D2 <- filter(free_only, year == "2015")
lm_free2015_only_totprod_D2 <- lm(tot_bacprod ~ D2, data = free_only_2015_D2) 
plot_totprod_freeonly_2015_D2 <- ggplot(free_only_2015_D2, aes(y = tot_bacprod, x = D2)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  ggtitle("2015 Prefiltered Free-Living") + 
  annotate("text", x = 30, y=70, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_free2015_only_totprod_D2)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_free2015_only_totprod_D2)$coefficients[,4][2]), digits = 4))) +
  theme(legend.position = c(0.12, 0.8), 
        legend.text = element_text(size = 10))


# Insignificant trend for the wholefree fraction only 
lm_wholefreeonly_totprod_D2 <- lm(tot_bacprod ~ D2, data = wholefree_only)
# Plot the relationship between wholefree and total production 
plot_totprod_wholefree_D2 <- ggplot(wholefree_only, aes(y = tot_bacprod, x = D2)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  ggtitle("WholeFree Only") + 
  annotate("text",  x = 40, y = 70, # For D2:  x = 40, y=5, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_wholefreeonly_totprod_D2)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_wholefreeonly_totprod_D2)$coefficients[,4][2]), digits = 4)))+
  theme(legend.position = c(0.12, 0.8), 
        legend.text = element_text(size = 10))


#### PARTICLE ASSOCIATED SAMPLES VS TOTAL BACTERIAL PRODUCTION 
# Is there a significant relationship?
lm_partonly_totprod_D2 <- lm(tot_bacprod ~ D2, data = part_only) 
# Plot the relationship
plot_totprod_part_D2 <- ggplot(part_only, aes(y = tot_bacprod, x = D2)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  geom_smooth(method = "lm", color = "black") + 
  ggtitle("2014 +2015 Prefiltered Particle Only") + 
  annotate("text", x = 55, y=5, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_partonly_totprod_D2)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_partonly_totprod_D2)$coefficients[,4][2]), digits = 4))) +
  theme(legend.position = c(0.12, 0.8), 
        legend.text = element_text(size = 10))


### Different particulate fractions 
## Particle (20-3um) fraction only 
part_only_2015 <- filter(part_only, year == "2015")
lm_part2015_only_totprod_D2 <- lm(tot_bacprod ~ D2, data = part_only_2015) 
plot_totprod_partonly_2015_D2 <- ggplot(part_only_2015, aes(y = tot_bacprod, x = D2)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  geom_smooth(method = "lm", color = "black") + 
  ggtitle("2015 20um Prefiltered Particle") + 
  annotate("text", x = 55, y=5, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_part2015_only_totprod_D2)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_part2015_only_totprod_D2)$coefficients[,4][2]), digits = 4))) +
  theme(legend.position = c(0.12, 0.8), 
        legend.text = element_text(size = 10))

### Whole Particle (3+ um) fraction only 
wholepart_only_2015 <- filter(wholepart_only, year == "2015")
lm_wholepart2015_only_totprod_D2 <- lm(tot_bacprod ~ D2, data = wholepart_only_2015) 
summary(lm_wholepart2015_only_totprod_D2)

# So this data has 2 outliers and robust regression is necessary 
#library(MASS)
# To put less weight in the outliers -> run robust regression
#rlm_wholepart2015_only_totprod_D2 <- rlm(tot_bacprod ~ D2, data = wholepart_only_2015) # To perform robust regression 
#summary(rlm_wholepart2015_only_totprod_D2)
#car::Anova(rlm_wholepart2015_only_totprod_D2)
#detach("package:MASS", unload=TRUE)
# TO calculate bootstrapped confiendence intervals on qq plot
#car::qqPlot(rlm_part_totprod_D2)
# To plot robust regression in ggplot do geom_smooth(method = "rlm")
#robust_wholepart2015_only_totprod_D2 <- lmRob(tot_bacprod ~ D2, data = wholepart_only_2015)
#summary(robust_wholepart2015_only_totprod_D2)

#### Dear marian, check out this website:  https://www.rdocumentation.org/packages/wle/versions/0.9-91/topics/wle.lm

plot_totprod_wholepartonly_D2 <- ggplot(wholepart_only_2015, aes(y = tot_bacprod, x = D2)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) +
  xlim(0, 100) +
  geom_smooth(method = "lm", color = "black") + 
  ggtitle("WholePart Only") + 
  annotate("text", x = 75, y=10, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_wholepart2015_only_totprod_D2)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_wholepart2015_only_totprod_D2)$coefficients[,4][2]), digits = 4))) +
  theme(legend.position = c(0.12, 0.8), 
        legend.text = element_text(size = 10))


## Put all the plots together
plot_D2_totprod <- plot_grid(plot_totprod_free_D2, plot_totprod_part_D2,
          labels = c("A", "B"), ncol = 2)
plot_D2_totprod

### Plot it all together for both of the free living fractions fractions 
plot_D2_totprod_FL_comparison <- plot_grid(plot_totprod_freeonly_2015_D2, plot_totprod_wholefree_D2,
          labels = c("A", "B"), ncol = 2)
plot_D2_totprod_FL_comparison

### Plot it all together for both of the particle associated fractions fractions 
plot_D2_totprod_PA_comparison <- plot_grid(plot_totprod_partonly_2015_D2, plot_totprod_wholepartonly_D2,
          labels = c("A", "B"), ncol = 2)
plot_D2_totprod_PA_comparison
```


# Is there a relationship between HNA cells per uL and Total Production?
```{r totalproduction_vs_HNA-LNA, fig.height=5, fig.width=14}
# nosed_meta_data # ALL SAMPLES 

# Is there a significant relationship between total cells per uL and total production?
lm_all_totprod_cells <- lm(tot_bacprod ~ cells_per_uL, data = nosed_meta_data)
# Plot the relationship between total cells per uL and total production
plot_totprod_vs_cells <- ggplot(nosed_meta_data, aes(x = cells_per_uL, y = tot_bacprod)) + 
  geom_point(aes(color = lakesite),  size = 3) +
  scale_color_manual(values = lakesite_colors) + 
  geom_smooth(method = "lm", color = "black") + 
  ggtitle("Total cells per uL") + 
  annotate("text", x = 9000, y=15, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_all_totprod_cells)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_all_totprod_cells)$coefficients[,4][2]), digits = 5))) +
  theme(legend.position = c(0.2, 0.7), legend.text = element_text(size = 10))


# Is there a significant relationship between HNA per uL and total production?
lm_all_totprod_HNA <- lm(tot_bacprod ~ HNA_per_uL, data = nosed_meta_data)
# Plot the relationship between HNA per uL and total production
plot_totprod_vs_HNA <- ggplot(nosed_meta_data, aes(x = HNA_per_uL, y = tot_bacprod)) + 
  geom_point(aes(color = lakesite),  size = 3) +
  scale_color_manual(values = lakesite_colors) + 
  geom_smooth(method = "lm", color = "black") + 
  ggtitle("HNA cells per uL") + 
  annotate("text", x = 3000, y=15, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_all_totprod_HNA)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_all_totprod_HNA)$coefficients[,4][2]), digits = 8))) +
  theme(legend.position = c(0.2, 0.67),  legend.text = element_text(size = 10))


# Is there a significant relationship between LNA per uL and total production?
lm_all_totprod_LNA <- lm(tot_bacprod ~ LNA_per_uL, data = nosed_meta_data)
# Plot the relationship between LNA per uL and total production
plot_totprod_vs_LNA <- ggplot(nosed_meta_data, aes(x = LNA_per_uL, y = tot_bacprod)) + 
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) + 
  ggtitle("LNA cells per uL") + 
  annotate("text", x = 3000, y=15, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_all_totprod_LNA)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_all_totprod_LNA)$coefficients[,4][2]), digits = 4))) +
  theme(legend.position = c(0.2, 0.67), legend.text = element_text(size = 10))


# Put all the plots together 
plots_totprod_vs_cellsHNALNA <- plot_grid(plot_totprod_vs_cells, plot_totprod_vs_HNA, plot_totprod_vs_LNA, 
          labels = c("A", "B", "C"), ncol = 3)
plots_totprod_vs_cellsHNALNA


df_hna_vs_lna <- nosed_meta_data %>%
  dplyr::select(HNA_per_uL, LNA_per_uL, norep_filter_name) %>%
  gather(cell_type, cells_per_uL, HNA_per_uL, LNA_per_uL)
  
  
df_hna_vs_lna$cell_type[df_hna_vs_lna$cell_type == "HNA_per_uL"] <- "HNA"
df_hna_vs_lna$cell_type[df_hna_vs_lna$cell_type == "LNA_per_uL"] <- "LNA"

ggplot(df_hna_vs_lna, aes(x = cell_type, y = cells_per_uL, color = cell_type, fill = cell_type)) + 
  geom_jitter(size = 3) + xlab("") + 
  geom_boxplot(alpha = 0.5) +
  theme(legend.position = c(0.15, 0.85)) +#
  scale_fill_manual(values = c("#F8BB0A", "#FF4E50")) +
  scale_color_manual(values = c("#F8BB0A", "#FF4E50")) 

#ggsave("Figures/HNA_LNA_vs_cellperUL.png", dpi = 500, width = 5, height = 4)
```




```{r HNA_vs_diversity,  fig.height=8, fig.width=10, echo = TRUE}
# Is there a relationship between HNA_percent and D2?
lm_free_D0_vs_HNApercent <- lm(D0 ~ HNA_percent, data = free_meta_data)
lm_free_D0chao_vs_HNApercent <- lm(D0_chao ~ HNA_percent, data = free_meta_data)
lm_free_D1_vs_HNApercent <- lm(D1 ~ HNA_percent, data = free_meta_data)
lm_free_D2_vs_HNApercent <- lm(D2 ~ HNA_percent, data = free_meta_data)


 HNA_vs_D0 <- ggplot(free_meta_data, aes(x = D0, y = HNA_percent)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) + 
  geom_smooth(method = "lm", color = "black") + 
  annotate("text", x = 2500, y=22, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_free_D0_vs_HNApercent)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_free_D0_vs_HNApercent)$coefficients[,4][2]), digits = 8))) +
  theme(legend.position = c(0.12, 0.8), legend.text = element_text(size = 10))


HNA_vs_chao <- ggplot(free_meta_data, aes(x = D0_chao, y = HNA_percent)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) + 
  geom_smooth(method = "lm", color = "black") + 
  annotate("text", x = 4000, y=20, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_free_D0chao_vs_HNApercent)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_free_D0chao_vs_HNApercent)$coefficients[,4][2]), digits = 8))) +
  theme(legend.position = c(0.12, 0.8), legend.text = element_text(size = 10))

HNA_vs_D1 <- ggplot(free_meta_data, aes(x = D1, y = HNA_percent)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) + 
  geom_smooth(method = "lm", color = "black") + 
  annotate("text", x = 120, y=22, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_free_D1_vs_HNApercent)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_free_D1_vs_HNApercent)$coefficients[,4][2]), digits = 8))) +
  theme(legend.position = c(0.12, 0.8), legend.text = element_text(size = 10))


HNA_vs_D2 <- ggplot(free_meta_data, aes(x = D2, y = HNA_percent)) +
  geom_point(aes(color = lakesite), size = 3) +
  scale_color_manual(values = lakesite_colors) + 
  geom_smooth(method = "lm", color = "black") + 
  annotate("text", x = 40, y=20, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_free_D2_vs_HNApercent)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_free_D2_vs_HNApercent)$coefficients[,4][2]), digits = 8))) +
  theme(legend.position = c(0.12, 0.8), legend.text = element_text(size = 10))


plot_grid(HNA_vs_D0, HNA_vs_chao, HNA_vs_D1, HNA_vs_D2,
          labels = c("A", "B", "C", "D"),
          ncol = 2, nrow = 2)



# Is there a relationship between LNA_percent and D2?
lm_free_D2_vs_LNApercent <- lm(D2 ~ LNA_percent, data = free_meta_data)

LNA_vs_D2 <- ggplot(free_meta_data, aes(x = D2, y = LNA_percent*100)) +
  geom_point(aes(color = lakesite, shape = season), size = 3) +
  scale_color_manual(values = lakesite_colors) + 
  ggtitle("LNA cells per uL") + 
  geom_smooth(method = "lm", color = "black") + 
  annotate("text", x = 48, y=80, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_free_D2_vs_LNApercent)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_free_D2_vs_LNApercent)$coefficients[,4][2]), digits = 8))) +
  theme(legend.position = c(0.15, 0.27), legend.text = element_text(size = 10))
         
HLNA_vs_D2 <- plot_grid(HNA_vs_D2, LNA_vs_D2, 
     labels = c("A", "B"), 
     ncol = 2)

# There is apparently a correlation between HNA_Percent and diversity of free living bacteria 
cor(free_meta_data$D2, free_meta_data$HNA_percent)
cor.test(free_meta_data$D2, free_meta_data$HNA_percent)

cor(free_meta_data$D1, free_meta_data$HNA_percent)
cor.test(free_meta_data$D1, free_meta_data$HNA_percent)

cor(free_meta_data$D0, free_meta_data$HNA_percent)
cor.test(free_meta_data$D0, free_meta_data$HNA_percent)

cor(free_meta_data$D0_chao, free_meta_data$HNA_percent)
cor.test(free_meta_data$D0_chao, free_meta_data$HNA_percent)
```



# Relationship between Chla and Total Production?
```{r chla-vs-totprod, echo = TRUE, fig.height=4, fig.width=5}
# Is there a relationship between CHLOROPHYLL A and Total Production?
lm_all_totprod_chla <- lm(tot_bacprod ~ Chl_Lab_ugperL, data = nosed_meta_data)
# Plot the relationship
ggplot(nosed_meta_data, aes(x = Chl_Lab_ugperL, y = tot_bacprod)) + 
  geom_point(size = 3) +
  geom_smooth(method = "lm") + 
  annotate("text", x = 6, y=75, 
           color = "black", fontface = "bold",
           label = paste("R2 =", round(summary(lm_all_totprod_chla)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_all_totprod_chla)$coefficients[,4][2]), digits = 4)))+
  theme(legend.position = c(0.12, 0.8), 
        legend.text = element_text(size = 10))
```

There appears to be *no relationship* between **chlorophyll a and total production**.



# Is there a relationship between diversity and *fractionated* bacterial production?
```{r create-fracprod-metadata, echo = TRUE}
# For 2015 data only 
meta_data <- data.frame(sample_data(otu_merged_musk_pruned)) 

all_top_2015_df <- dplyr::filter(meta_data, limnion == "Top" & 
                              norep_filter_name != "MOTHJ715" & 
                              year == "2015")

wholefree_2015_df <- dplyr::filter(meta_data, fraction == "WholeFree" & 
                              limnion == "Top" & 
                              norep_filter_name != "MOTHJ715" & 
                              year == "2015")

wholeparticle_2015_df <- dplyr::filter(meta_data, fraction == "WholePart" & 
                                  limnion == "Top" &
                                  norep_filter_name != "MOTHJ715" & 
                                  year == "2015")
```


```{r fractional_production_vs_simps_even, echo = TRUE, fig.height=4, fig.width=5}
# Can fractional production be predicted by phenoflow D2 INVERSE SIMPSON? 
# FREE LIVING
free_otu_simps_even_stats <- lm(frac_bacprod ~ D2/D0, data = wholefree_2015_df)
# PARTICLE
part_otu_simps_even_stats <- lm(frac_bacprod ~ D2/D0, data = wholeparticle_2015_df)

# Plot D2 INVERSE SIMPSON
ggplot(all_top_2015_df, aes(x=D2/D0, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) + 
  geom_errorbar(aes(ymin = frac_bacprod - SD_frac_bacprod, ymax = frac_bacprod + SD_frac_bacprod)) + 
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  ylab("Production (μgC/L/hr)") + xlab("Simpson's Evenness (D2/D0)") +
  theme(legend.position=c(0.85,0.9), legend.title=element_blank()) +
  annotate("text", x = 0.04, y=60, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(free_otu_simps_even_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(free_otu_simps_even_stats)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 0.06, y=35, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(part_otu_simps_even_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(part_otu_simps_even_stats)$coefficients[,4][2]), digits = 4)))
```




```{r fractional_production_vs_diversity, echo = TRUE, fig.width=10, fig.height = 10}
# Can fractional production be predicted by phenoflow D0 observed richness? 
# FREE LIVING
free_otu_D0_stats <- lm(frac_bacprod ~ D0, data = wholefree_2015_df)
#summary(free_otu_D0_stats)

# PARTICLE
part_otu_D0_stats <- lm(frac_bacprod ~ D0, data = wholeparticle_2015_df)
#summary(part_otu_D0_stats)

# Plot D0 Observed Richness
otu_pheno_D0 <- ggplot(all_top_2015_df, aes(x=D0, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) + 
  geom_errorbarh(aes(xmin = D0 - D0_SD, xmax = D0 + D0_SD)) + 
  geom_errorbar(aes(ymin = frac_bacprod - SD_frac_bacprod, ymax = frac_bacprod + SD_frac_bacprod)) + 
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  ylab("Production (μgC/L/hr)") + xlab("Observed Richness (D0)") +
  geom_smooth(data=subset(all_top_2015_df, fraction == "WholePart"), method='lm') + 
  theme(legend.position=c(0.15,0.9), legend.title=element_blank()) +
  annotate("text", x = 1000, y=35, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(free_otu_D0_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(free_otu_D0_stats)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 1600, y=8, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(part_otu_D0_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(part_otu_D0_stats)$coefficients[,4][2]), digits = 4)))


# Can fractional production be predicted by phenoflow chao1? 
# FREE LIVING
free_otu_D0chao_stats <- lm(frac_bacprod ~ D0_chao, data = wholefree_2015_df)
#summary(free_otu_D0chao_stats)

# PARTICLE
part_otu_D0chao_stats <- lm(frac_bacprod ~ D0_chao, data = wholeparticle_2015_df)
#summary(part_otu_D0chao_stats)
 
# Plot  chao1
otu_pheno_D0chao <- ggplot(all_top_2015_df, aes(x=D0_chao, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) + 
  geom_errorbarh(aes(xmin = D0_chao - D0_chao_sd, xmax = D0_chao + D0_chao_sd)) + 
  geom_errorbar(aes(ymin = frac_bacprod - SD_frac_bacprod, ymax = frac_bacprod + SD_frac_bacprod)) + 
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  ylab("Production (μgC/L/hr)") + xlab("Chao1 (D0)") +
  geom_smooth(data=subset(all_top_2015_df, fraction == "WholePart"), method='lm') + 
  theme(legend.position=c(0.15,0.9), legend.title=element_blank()) +
  annotate("text", x = 2000, y=35, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(free_otu_D0chao_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(free_otu_D0chao_stats)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 2700, y=8, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(part_otu_D0chao_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(part_otu_D0chao_stats)$coefficients[,4][2]), digits = 4)))


# Can fractional production be predicted by phenoflow D1 SHANNON ENTROPY? 
# FREE LIVING
free_otu_D1_stats <- lm(frac_bacprod ~ D1, data = wholefree_2015_df)
#summary(free_otu_D1_stats)

# PARTICLE
part_otu_D1_stats <- lm(frac_bacprod ~ D1, data = wholeparticle_2015_df)
#summary(part_otu_D1_stats)

# Plot D1 SHANNON ENTROPY
otu_pheno_D1 <- ggplot(all_top_2015_df, aes(x=D1, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) + 
  geom_errorbarh(aes(xmin = D1 - D1_sd, xmax = D1 + D1_sd)) + 
  geom_errorbar(aes(ymin = frac_bacprod - SD_frac_bacprod, ymax = frac_bacprod + SD_frac_bacprod)) + 
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  ylab("Production (μgC/L/hr)") + xlab("Shannon Entropy (D1)") +
  geom_smooth(data=subset(all_top_2015_df, fraction == "WholePart"), method='lm') + 
  theme(legend.position=c(0.85,0.9), legend.title=element_blank()) +
  annotate("text", x = 175, y=35, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(free_otu_D1_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(free_otu_D1_stats)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 275, y=7, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(part_otu_D1_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(part_otu_D1_stats)$coefficients[,4][2]), digits = 4)))


# Can fractional production be predicted by phenoflow D2 INVERSE SIMPSON? 
# FREE LIVING
lm_free_otu_D2_stats <- lm(frac_bacprod ~ D2, data = wholefree_2015_df)
#summary(free_otu_D2_stats)

# PARTICLE
lm_part_otu_D2_stats <- lm(frac_bacprod ~ D2, data = wholeparticle_2015_df)
#summary(part_otu_D2_stats)

# To put less weight in the outliers -> run robust regression
#rlm_part_otu_D2_stats <- rlm(frac_bacprod ~ D2, data = wholeparticle_2015_df) # To perform robust regression 
#summary(rlm_part_otu_D2_stats)
#car::Anova(rlm_part_otu_D2_stats)
# To plot robust regression in ggplot do geom_smooth(method = "rlm")


# Plot D2 INVERSE SIMPSON
otu_pheno_D2 <- ggplot(all_top_2015_df, aes(x=D2, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) + 
  geom_errorbarh(aes(xmin = D2 - D2_sd, xmax = D2 + D2_sd)) +
  geom_errorbar(aes(ymin = frac_bacprod - SD_frac_bacprod, ymax = frac_bacprod + SD_frac_bacprod)) + 
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  ylab("Production (μgC/L/hr)") + xlab("Inverse Simpson (D2)") +
  geom_smooth(data=subset(all_top_2015_df, fraction == "WholePart"), method='lm') + 
  theme(legend.position=c(0.85,0.9), legend.title=element_blank()) +
  annotate("text", x = 40, y=45, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(lm_free_otu_D2_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_free_otu_D2_stats)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 50, y=25, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(lm_part_otu_D2_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(lm_part_otu_D2_stats)$coefficients[,4][2]), digits = 4)))

otu_phenoflow <- plot_grid(otu_pheno_D0, otu_pheno_D0chao, otu_pheno_D1, otu_pheno_D2, 
          labels = c("A", "B", "C", "D"), 
          align = "h", nrow = 2, ncol = 2)
otu_phenoflow

#ggsave("../Figures/fracprod_vs_diversity.png", otu_phenoflow, dpi = 600, units = "in", width = 10, height = 8)
```



# Phylum production analysis
```{r phylum-div-prod, echo = TRUE}
# Michelle Berry's subsetting function, similar to phyloseq::subset_taxa, except taxa can 
# be passed as arguments within functions without weird environment errors
#
# Args:
#   physeq: a phyloseq object
#   taxrank: taxonomic rank to filter on
#   taxa: a vector of taxa groups to filter on
#
# Returns: 
#   a phyloseq object subsetted to the x taxa in taxrank
my_subset_taxa <- function(physeq, taxrank, taxa) {
  physeq_tax_sub <- tax_table(physeq)[tax_table(physeq)[ , taxrank] %in% taxa, ]
  tax_table(physeq) <- physeq_tax_sub
  return(physeq)
}

## Scale phyloseq objects of whole free and whole particle:
# scale_otu_merged_musk_wholefree_top
# scale_otu_merged_musk_wholepart_top


# Initialize parameters
trials <- 100
min_lib <- min(sample_sums(scale_otu_merged_musk_wholepart_top)) # Depth we are rarefying to

# Groups to estimate alpha diversity for 
mytaxa <- c("Acidobacteria", "Actinobacteria", "Alphaproteobacteria", "Armatimonadetes",
            "Bacteroidetes", "Betaproteobacteria", "Chlorobi", "Chloroflexi",
            "Cyanobacteria","Deltaproteobacteria", "Gammaproteobacteria", 
            "Planctomycetes","Verrucomicrobia")
names(mytaxa) <- mytaxa

# Taxonomic ranks of mytaxa
mytaxa_taxrank <- c("Phylum", "Phylum", "Phylum", "Phylum", "Phylum", "Phylum", 
                      "Phylum", "Phylum", "Phylum", "Phylum", "Phylum", "Phylum", "Phylum")
names(mytaxa_taxrank) <- mytaxa

# Data frame to hold alpha diversity estimates over trials
alphadiv_df <- data.frame(matrix(nrow = nsamples(scale_otu_merged_musk_wholepart_top), ncol = trials))

# Initialize empty df's for richness and evenness of all taxa in mytaxa
richness <- lapply(mytaxa, function(x) {return(alphadiv_df)} )
invsimps <- lapply(mytaxa, function(x) {return(alphadiv_df)} )

alphadiv_list <- list(richness = richness, invsimps = invsimps)


# Set the seed for reproducibility
set.seed(777)

# Run trials to subsample and estimate diversity
for (i in 1:100) {
  
  # Subsample
  rarefied_physeq <- rarefy_even_depth(scale_otu_merged_musk_wholepart_top, sample.size = min_lib, verbose = FALSE, replace = TRUE)
  
  # Generate alpha-diversity estimates for each taxonomic group
  for (t in mytaxa) {

    # Subset the taxa
    physeq_sub <- my_subset_taxa(physeq = rarefied_physeq, 
                                 taxrank = mytaxa_taxrank[t], 
                                 taxa = t)
    
    # Calculate observed richness for that group and store value in a df column
    richness <- estimate_richness(physeq_sub, measures = "Observed")[ ,1]
    alphadiv_list$richness[[t]][ ,i] <- richness
    
    # Calculate observed invsimps for that group and store value in a df column
    invsimps <- estimate_richness(physeq_sub, measures = "InvSimpson")[ ,1]
    alphadiv_list$invsimps[[t]][ ,i] <- invsimps

  }
}

# Calculate the means of richness and inverse simpson from the 100 trials
alphadiv_est <- lapply(alphadiv_list, function(div_measure) {
    lapply(div_measure, function(taxa_group) {
        alpha_mean <- rowMeans(taxa_group)
        return(alpha_mean)
    })  
})

# Convert alphadiv_est richness and simpson's E lists into wide data frames
l <- lapply(alphadiv_est, function(x) {
  # convert from list to data.frame
  est_df <- plyr::ldply(.data = x, .fun = data.frame)
  names(est_df) <- c("Taxa", "Diversity")
  
  # Add in SampleID column and spread to wide format
  r <- est_df %>%
    mutate(norep_filter_name = rep(sample_names(scale_otu_merged_musk_wholepart_top), length(mytaxa)))
  return(r)
})

# Merge sample metadata with these estimates
merge_dat <- data.frame(sample_data(scale_otu_merged_musk_wholepart_top)) %>%
  dplyr::select(norep_filter_name, Chl_Lab_ugperL, TP_ugperL, pH, frac_bacprod, NH3_mgperL, NO3_mgperL,
         lakesite, fraction, season) 

# Create a df with a "Diversity" column that includes richness and inv. simpson,
# and log-chl a values from erie sample_data
alpha_comb1 <- l$richness %>% 
  left_join(y = l$invsimps, by = c("Taxa", "norep_filter_name")) %>%   # Join the richness and inv_simp df's
  rename(Richness = Diversity.x, Inverse_Simpson = Diversity.y) 

alpha_comb_final <- alpha_comb1 %>%
  left_join(merge_dat, by = "norep_filter_name") %>%                  # Join with merged nutrient data
  gather(key = "Alphadiv", value = "Estimate", Richness, Inverse_Simpson) 
```


```{r plot_phylum_diversity, fig.width=8, fig.height=20}
ggplot(alpha_comb_final, aes(x = Estimate, y =frac_bacprod)) + 
  geom_point() +
  geom_smooth(method = "lm") + 
  facet_grid(Taxa~Alphadiv, scales = "free_x")

alpha_comb_final$Alphadiv <- factor(alpha_comb_final$Alphadiv,
                                    levels=c("Richness", "Inverse_Simpson"),
                                    ordered=TRUE)

#ggsave("../Figures/phyla_alphadiv.png", dpi = 600, width = 6, height = 20)
```




```{r specific_phyla, fig.width=10, fig.height = 4, echo = TRUE}
# CYANOS 
cyanos <- filter(alpha_comb_final, Taxa == "Cyanobacteria")
summary(lm(frac_bacprod ~ Estimate, data = filter(cyanos, Alphadiv == "Richness")))
summary(lm(frac_bacprod ~ Estimate, data = filter(cyanos, Alphadiv == "Inverse_Simpson")))
# Plot it 
ggplot(cyanos, aes(x = Estimate, y =frac_bacprod)) + 
  geom_point(size = 3) +
  #geom_smooth(method = "lm") + 
  facet_grid(Taxa~Alphadiv, scales = "free_x")



# ACTINOBACTERIA
actinos <- filter(alpha_comb_final, Taxa == "Actinobacteria")
summary(lm(frac_bacprod ~ Estimate, data = filter(actinos, Alphadiv == "Richness")))
summary(lm(frac_bacprod ~ Estimate, data = filter(actinos, Alphadiv == "Inverse_Simpson")))
# Plot it 
ggplot(actinos, aes(x = Estimate, y =frac_bacprod)) + 
  geom_point(size = 3) +
  #geom_smooth(method = "lm") + 
  facet_grid(Taxa~Alphadiv, scales = "free_x")


# BACTEROIDETES
bacteroidetes <- filter(alpha_comb_final, Taxa == "Bacteroidetes")
summary(lm(frac_bacprod ~ Estimate, data = filter(bacteroidetes, Alphadiv == "Richness")))
summary(lm(frac_bacprod ~ Estimate, data = filter(bacteroidetes, Alphadiv == "Inverse_Simpson")))
# Plot it 
ggplot(bacteroidetes, aes(x = Estimate, y =frac_bacprod)) + 
  geom_point(size = 3) +
  geom_smooth(method = "lm", data = filter(bacteroidetes, Alphadiv == "Richness")) + 
  facet_grid(Taxa~Alphadiv, scales = "free_x")


# VERRUCOMICROBIA
verruco <- filter(alpha_comb_final, Taxa == "Verrucomicrobia")
summary(lm(frac_bacprod ~ Estimate, data = filter(verruco, Alphadiv == "Richness")))
summary(lm(frac_bacprod ~ Estimate, data = filter(verruco, Alphadiv == "Inverse_Simpson")))
# Plot it 
ggplot(verruco, aes(x = Estimate, y =frac_bacprod)) + 
  geom_point(size = 3) +
  geom_smooth(method = "lm") + 
  facet_grid(Taxa~Alphadiv, scales = "free_x")


# PLANCTOMYCETES
planctos <- filter(alpha_comb_final, Taxa == "Planctomycetes")
summary(lm(frac_bacprod ~ Estimate, data = filter(planctos, Alphadiv == "Richness")))
summary(lm(frac_bacprod ~ Estimate, data = filter(planctos, Alphadiv == "Inverse_Simpson")))
# Plot it 
ggplot(planctos, aes(x = Estimate, y =frac_bacprod)) + 
  geom_point(size = 3) +
  geom_smooth(method = "lm", data = filter(planctos, Alphadiv == "Richness")) + 
  facet_grid(Taxa~Alphadiv, scales = "free_x")


# BETAPROTEOS
betaprots <- filter(alpha_comb_final, Taxa == "Betaproteobacteria")
summary(lm(frac_bacprod ~ Estimate, data = filter(betaprots, Alphadiv == "Richness")))
summary(lm(frac_bacprod ~ Estimate, data = filter(betaprots, Alphadiv == "Inverse_Simpson")))
# Plot it 
ggplot(betaprots, aes(x = Estimate, y =frac_bacprod)) + 
  geom_point(size = 3) +
  geom_smooth(method = "lm", data = filter(betaprots, Alphadiv == "Richness")) + 
  facet_grid(Taxa~Alphadiv, scales = "free_x")


# ALPHAPROTS
alphaprots <- filter(alpha_comb_final, Taxa == "Alphaproteobacteria")
summary(lm(frac_bacprod ~ Estimate, data = filter(alphaprots, Alphadiv == "Richness")))
summary(lm(frac_bacprod ~ Estimate, data = filter(alphaprots, Alphadiv == "Inverse_Simpson")))
# Plot it 
ggplot(alphaprots, aes(x = Estimate, y =frac_bacprod)) + 
  geom_point(size = 3) +
  geom_smooth(method = "lm", data = filter(alphaprots, Alphadiv == "Richness")) + 
  facet_grid(Taxa~Alphadiv, scales = "free_x")
```




# How do the samples relate to each other? 
## PCOA
```{r PCOA, fig.width=16, fig.height=8}
# Calculate the bray-curtis dissimilarity
# For ALL samples
water_otu_pcoa_bray <- ordinate(
  physeq = scale_otu_merged_musk_nosed,
  method = "PCoA",
  distance = "bray")

# Plot the ordination of ALL samples 
water_pcoa_bray <- plot_ordination(
  physeq = scale_otu_merged_musk_nosed,
  ordination = water_otu_pcoa_bray,
  color = "fraction",
  shape = "season",
  title = "Bray: All") + 
  scale_colour_manual(values = fraction_colors) +
  geom_point(size=5, alpha = 0.5) 



####   FREE LIVING SAMPLES 
# Calculate the bray-curtis dissimilarity
# For ALL samples
free_otu_pcoa_bray <- ordinate(
  physeq = otu_merged_musk_pruned_free,
  method = "PCoA",
  distance = "bray")

# Plot the ordination of ALL samples 
free_pcoa_bray <- plot_ordination(
  physeq = otu_merged_musk_pruned_free,
  ordination = free_otu_pcoa_bray,
  color = "fraction",
  shape = "season",
  title = "Bray: FL") + 
  scale_colour_manual(values = fraction_colors) +
  geom_point(size=5, alpha = 0.5) 


####   PARTICLE ASSOCIATED SAMPLES  
# Calculate the bray-curtis dissimilarity
# For ALL samples
part_otu_pcoa_bray <- ordinate(
  physeq = otu_merged_musk_pruned_particle,
  method = "PCoA",
  distance = "bray")

# Plot the ordination of ALL samples 
part_pcoa_bray <- plot_ordination(
  physeq = otu_merged_musk_pruned_particle,
  ordination = part_otu_pcoa_bray,
  color = "fraction",
  shape = "season",
  title = "Bray: PA") + 
  scale_colour_manual(values = fraction_colors) +
  geom_point(size=5, alpha = 0.5) 

############################################
# Calculate the SORENSEN dissimilarity
# For ALL samples
water_otu_pcoa_soren <- ordinate(
  physeq = scale_otu_merged_musk_nosed,
  method = "PCoA",
  distance = distance(scale_otu_merged_musk_nosed, method = "bray", binary = TRUE, type = "samples"))

# Plot the ordination of ALL samples 
water_pcoa_soren <- plot_ordination(
  physeq = scale_otu_merged_musk_nosed,
  ordination = water_otu_pcoa_soren,
  color = "fraction",
  shape = "season",
  title = "Sorensen: All") + 
  scale_colour_manual(values = fraction_colors) +
  geom_point(size=5, alpha = 0.5) 



############################################
####   FREE LIVING SAMPLES 
# Calculate the SORENSEN dissimilarity 
free_otu_pcoa_soren <- ordinate(
  physeq = otu_merged_musk_pruned_free,
  method = "PCoA",
  distance = distance(otu_merged_musk_pruned_free, method = "bray", binary = TRUE, type = "samples"))

# Plot the ordination of FREE LIVING samples 
free_pcoa_soren <- plot_ordination(
  physeq = otu_merged_musk_pruned_free,
  ordination = free_otu_pcoa_soren,
  color = "fraction",
  shape = "season",
  title = "Sorensen: FL") + 
  scale_colour_manual(values = fraction_colors) +
  geom_point(size=5, alpha = 0.5) 

############################################
####   PARTICLE ASSOCIATED SAMPLES  
# Calculate the SORENSEN dissimilarity
part_otu_pcoa_soren <- ordinate(
  physeq = otu_merged_musk_pruned_particle,
  method = "PCoA",
  distance = distance(otu_merged_musk_pruned_particle, method = "bray", binary = TRUE, type = "samples"))

# Plot the ordination of PARTICLE ASSOCIATED  samples 
part_pcoa_soren <- plot_ordination(
  physeq = otu_merged_musk_pruned_particle,
  ordination = part_otu_pcoa_soren,
  color = "fraction",
  shape = "season",
  title = "Sorensen:PA") + 
  scale_colour_manual(values = fraction_colors) +
  geom_point(size=5, alpha = 0.5) 

#  All the plots 
all_the_pcoas <- plot_grid(water_pcoa_bray, free_pcoa_bray, part_pcoa_bray, 
          water_pcoa_soren, free_pcoa_soren, part_pcoa_soren, 
          labels = c("A", "B", "C", "D", "E", "F"), 
          nrow = 2, ncol = 3)

all_the_pcoas
#ggsave("../Figures/pcoas.png", all_the_pcoas, 
#       dpi = 600, width = 16, height = 8)

```



#  Stacked Bar plots for taxonomic structure
## OTU  Phylum Stacked Bar 
```{r otu_stacked_bar, eval = FALSE}
musk_scaled_otu_merged_pruned <- subset_samples(scale_otu_merged_musk_pruned, project == "Muskegon_Lake" & 
                                           year == "2015" &
                                           limnion == "Top" &
                                           fraction %in% c("WholePart", "WholeFree", "Free"))

# Fix month levels in sample_data
sample_data(musk_scaled_otu_merged_pruned)$fraction <- factor(
  sample_data(musk_scaled_otu_merged_pruned)$fraction, 
  levels = c("WholePart", "WholeFree", "Free"))

sample_data(musk_scaled_otu_merged_pruned)$lakesite <- factor(
  sample_data(musk_scaled_otu_merged_pruned)$lakesite, 
  levels = c("MOT", "MDP", "MBR", "MIN"))



musk_scaled_otu_merged_pruned_phylum <- musk_scaled_otu_merged_pruned %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa
  arrange(Phylum)                                      # Sort data frame alphabetically by phylum


# Plot 
otu_stack_bar <- ggplot(musk_scaled_otu_merged_pruned_phylum, aes(x = season, y = Abundance, fill = Phylum)) + 
  facet_grid(fraction~lakesite) +
  geom_bar(stat = "identity", color = "black") +
  scale_fill_manual(values = phylum_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  ggtitle("Phylum Composition of 2015 Muskegon Lake \n Bacterial Communities by Sampling Site") +
  theme(strip.background = element_rect(fill = "white"),
        axis.text.x = element_text(colour = "black", angle=45, hjust = 1, vjust = 1))
otu_stack_bar
#ggsave("Figures/otu_phylum_stackedbar.png", otu_stack_bar, dpi = 600, units = "in", width = 12, height = 8)
```




# Environmental Conditions 
```{r, eval = FALSE}
prod_data <- data.frame(sample_data(musk_scaled_otu_merged_pruned))

ggplot(prod_data, aes(x = season, y = tot_bacprod, color = lakesite, group = lakesite, linetype = lakesite)) + 
  geom_point(size = 3.5) +   geom_line(size = 2) +
  ylab("Total Bacterial\n Production") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1))

#### Plot many environ variables
tot_prod_plot <- ggplot(prod_data, aes(x = season, y = tot_bacprod, color = lakesite, group = lakesite)) + 
  facet_grid(.~lakesite) +
  geom_point(size = 3.5) + geom_line() +
  ylab("Total Bacterial\n Production") +
  theme(strip.background = element_rect(fill = "white"),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1),
        legend.position="none")

chla_plot <- ggplot(prod_data, aes(x = season, y = Chl_Lab_ugperL, color = lakesite, group = lakesite)) + 
  facet_grid(.~lakesite) +
  geom_point(size = 3.5) + geom_line() +
  ylab("Chlorophyll a") +
  theme(strip.background = element_rect(fill = "white"),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1),
        legend.position="none")

ph_plot <- ggplot(prod_data, aes(x = season, y = pH, color = lakesite, group = lakesite)) + 
  facet_grid(.~lakesite) +
  geom_point(size = 3.5) + geom_line() +
  ylab("pH") +
  theme(strip.background = element_rect(fill = "white"),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1),
        legend.position="none")

tp_plot <- ggplot(prod_data, aes(x = season, y = TP_ugperL, color = lakesite, group = lakesite)) + 
  facet_grid(.~lakesite) +
  geom_point(size = 3.5) + geom_line() +
  ylab("TP_ugperL") +
  theme(strip.background = element_rect(fill = "white"),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1),
        legend.position="none")

no3_plot <- ggplot(prod_data, aes(x = season, y = NO3_mgperL, color = lakesite, group = lakesite)) + 
  facet_grid(.~lakesite) +
  geom_point(size = 3.5) + geom_line() +
  ylab("NO3_mgperL") +
  theme(strip.background = element_rect(fill = "white"),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1),
        legend.position="none")

nh3_plot <- ggplot(prod_data, aes(x = season, y = NH3_mgperL, color = lakesite, group = lakesite)) + 
  facet_grid(.~lakesite) +
  geom_point(size = 3.5) + geom_line() +
  ylab("NH3_mgperL") +
  theme(strip.background = element_rect(fill = "white"),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1),
        legend.position="none")

environ_seasons <- plot_grid(tot_prod_plot, chla_plot, ph_plot, tp_plot, no3_plot, nh3_plot,
          labels = c("A", "B", "C", "D", "E", "F"), 
          align = "h", nrow = 6, ncol = 1)
environ_seasons

#ggsave("Figures/environ_seasons.png", environ_seasons, dpi = 600, units = "in", width = 12, height = 12)
```






# Calculate the Absolute abundances!
```{r prepare-peter-physeq, eval = FALSE}
### MODIFY TO ABSOLUTE ABUNDANCES!

# First let's calculate the proportion
peter_physeq <- free_sample_physeq 
rel_abund_physeq <- peter_physeq %>%
  transform_sample_counts(function(x) {x/sum(x)} ) # Transform to rel. abundance

# Multiply the proportion by the total number of cells
relabund_otu <- otu_table(rel_abund_physeq)
absabund_counts_otu <- relabund_otu * sample_data(peter_physeq)$cells_per_uL
stopifnot(rowSums(absabund_counts_otu) == sample_data(peter_physeq)$cells_per_uL) # Sanity check
otu_table(peter_physeq) <- absabund_counts_otu # Reassign the otu_table to be the absolute counts
otu_table(peter_physeq)[1:5,1:15]

## From counts to absolute abundance proprtions 
absabund_proportion_otu <- peter_physeq %>%
  transform_sample_counts(function(x) {x/sum(x)} ) # Transform to rel. abundance
otu_table(absabund_proportion_otu)[1:5,1:15] == otu_table(rel_abund_physeq)[1:5,1:15] # Relative and absolute proportions are the same!

# Final otu data frames
peter_otu_counts <- absabund_counts_otu # OTU COUNTS TABLE
peter_otu_proportion <- otu_table(absabund_proportion_otu) # OTU PROPORTION TABLE

# Write the files 
#write.table(peter_otu_counts, "peter/muskegon_absolute_otu_counts.tsv", sep = "\t")
#write.table(peter_otu_proportion, "peter/muskegon_otu_proportions.tsv", sep = "\t")

#write.xlsx(x = peter_otu_counts, file = "peter/muskegon_absolute_otu_counts.xlsx")
#write.xlsx(x = peter_otu_proportion, file = "peter/muskegon_otu_proportions.xlsx")

HNA_prod3 <- HNA_prod2 %>%
  select(norep_filter_name, lakesite, limnion, year, season,
         tot_bacprod,
         cells_per_uL, HNA_per_uL, HNA_percent, LNA_per_uL, LNA_percent) %>%
  rename(depth = limnion)

row.names(HNA_prod3) = NULL

#write.table(HNA_prod3, "peter/sample_data.tsv", sep = "\t")
#write.xlsx(x = HNA_prod3, file = "peter/sample_data.xlsx")


tot_bacprod_plot1 <- ggplot(HNA_prod3, aes(y = tot_bacprod, x = lakesite, color = lakesite, fill = lakesite)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.9)) +
  scale_color_manual(values = lakesite_colors) +
  scale_fill_manual(values = lakesite_colors)


tot_bacprod_plot2 <- ggplot(HNA_prod3, aes(y = tot_bacprod, x = lakesite, color = year, fill = year)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.9)) +
  scale_color_manual(values = year_colors) +
  scale_fill_manual(values = year_colors)


tot_bacprod_plot3 <- ggplot(HNA_prod3, aes(y = tot_bacprod, x = lakesite, color = season, fill = season)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.9)) +
  scale_color_manual(values = season_colors) +
  scale_fill_manual(values = season_colors)

a_plot1 <- ggplot(HNA_prod3, aes(y = HNA_percent, x = lakesite, color = lakesite, fill = lakesite)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.9)) +
  scale_color_manual(values = lakesite_colors) +
  scale_fill_manual(values = lakesite_colors)


a_plot2 <- ggplot(HNA_prod3, aes(y = HNA_percent, x = lakesite, color = year, fill = year)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.9)) +
  scale_color_manual(values = year_colors) +
  scale_fill_manual(values = year_colors)


a_plot3 <- ggplot(HNA_prod3, aes(y = HNA_percent, x = lakesite, color = season, fill = season)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.9)) +
  scale_color_manual(values = season_colors) +
  scale_fill_manual(values = season_colors)


tot_bacprod_plotz <- plot_grid(tot_bacprod_plot1, tot_bacprod_plot2, tot_bacprod_plot3, 
           labels = c("A", "B", "C"), 
           align = "h", nrow = 3, ncol = 1)

#ggsave("Figures/tot_bacprod_factors.png", tot_bacprod_plotz, dpi = 600, units = "in", width = 5, height = 6)


# Write a season table for peter
season_df <- HNA_prod3 %>%
  select(norep_filter_name, season)
write.table(season_df, "peter/sample_season_data.tsv", sep = "\t", row.names = FALSE)



a_plotz <- plot_grid(tot_bacprod_plot1, a_plot1,tot_bacprod_plot2, a_plot2, tot_bacprod_plot3, a_plot3,
          labels = c("A", "B", "C", "D", "E", "F"), 
          align = "h", nrow = 3, ncol = 2)


ggplot(HNA_prod3, aes(y = HNA_percent, x = lakesite, color = lakesite, fill = lakesite)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.6)) +
  scale_color_manual(values = lakesite_colors) +
  scale_fill_manual(values = lakesite_colors)


```


### HNA VS LNA 
```{r, eval = FALSE}
####  PRODUCTION VS HNA

ggplot(HNA_prod, aes(x = Site, y = tot_bacprod, fill = Season, shape = Site)) + ggtitle("Surface samples only") +
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 


r1 <- ggplot(HNA_prod, aes(x = HNA_per_uL, y = tot_bacprod)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site), show.legend = FALSE) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

r2 <- ggplot(HNA_prod, aes(x = HNA_percent, y = tot_bacprod)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site), show.legend = FALSE) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

r3 <- ggplot(HNA_prod, aes(x = cells_per_uL, y = tot_bacprod)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site)) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

r4 <-ggplot(HNA_prod, aes(x = Nuc_acid_ratio, y = tot_bacprod)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site)) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

r5 <- ggplot(HNA_prod, aes(x = LNA_per_uL, y = tot_bacprod)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site), show.legend = FALSE) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

r6 <- ggplot(HNA_prod, aes(x = LNA_percent, y = tot_bacprod)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site), show.legend = FALSE) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 


rr <- plot_grid(r1, r5, r3, r2, r6, r4, 
          labels = c("A", "B", "C", "D", "E", "F"), 
          align = "h", nrow = 2, ncol = 3)

#ggsave("Figures/HNA_LNA_vs_production.png", rr, dpi = 600, units = "in", width = 14, height = 8)



##################################################################################################################################
##################################################################################################################################
##################################################################################################################################


a <- ggplot(HNA_prod, aes(x = HNA_per_uL, y = attached_bac)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site)) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) +
  theme(legend.position = c(0.15, 0.75))

b <- ggplot(HNA_prod, aes(x = HNA_per_uL, y = perc_attached_abund)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site)) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) +
  theme(legend.position = c(0.15, 0.75))

ab <- plot_grid(a, b,
          labels = c("A", "B"), 
          align = "h", nrow = 1, ncol = 2)

#ggsave("Figures/HNA_vs_attached_bac.png", ab, dpi = 600, units = "in", width = 10, height = 6)




ggplot(HNA_prod, aes(x = cells_per_uL, y = total_bac_abund/1000)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site)) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) +
  theme(legend.position = c(0.2, 0.75))


##################################################################################################################################
##################################################################################################################################
##################################################################################################################################
# Plot D0 Observed Richness
d0 <- ggplot(HNA_prod, aes(x=D0, y=tot_bacprod)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = D0 - D0.sd, xmax = D0 + D0.sd), width = 0.2) + 
  ggtitle("Phenotypic Div: Phenoflow") +
  ylab("Total Production (μgC/L/hr)") + xlab("Observed Richness (D0)") +
  theme(legend.position=c(0.15,0.9), legend.title=element_blank()) +
  geom_smooth(method = "lm", color = "black") + 
  scale_x_continuous(limits = c(13800, 16300)) 


d1 <- ggplot(HNA_prod, aes(x=D1, y=tot_bacprod)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = D1 - D1.sd, xmax = D1 + D1.sd), width = 0.2) + 
  ggtitle("Phenotypic Div: Phenoflow") +
  ylab("Total Production (μgC/L/hr)") + xlab("Shannon Entropy (D1)") +
  theme(legend.position=c(0.15,0.9), legend.title=element_blank()) +
  geom_smooth(method = "lm", color = "black") +
  scale_x_continuous(limits = c(3000, 3700)) 

d2 <- ggplot(HNA_prod, aes(x=D2, y=tot_bacprod)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = D2 - D2.sd, xmax = D2 + D2.sd), width = 0.2) + 
  ggtitle("Phenotypic Div: Phenoflow") +
  ylab("Total Production (μgC/L/hr)") + xlab("Inverse Simpson (D2)") +
  theme(legend.position=c(0.15,0.9), legend.title=element_blank()) +
  geom_smooth(method = "lm", color = "black") + 
  scale_x_continuous(limits = c(1850, 2300)) 


dd <- plot_grid(d0, d1, d2,
          labels = c("A", "B", "C"), 
          align = "h", nrow = 1, ncol = 3)

ggsave("Figures/phenotypic_div_vs_total_production.png", dd, dpi = 600, units = "in", width = 10, height = 6)

##################################################################################################################################
##################################################################################################################################
##################################################################################################################################
###  HNA PER uL
limits_musk <- c(600,3700)

surface_HNA <- ggplot(filter(flow_cy, Depth == "Surface"), 
       aes(x = Site, y = HNA_per_uL, fill = Season, shape = Site)) + ggtitle("Surface samples only") +
  scale_y_continuous(limits = limits_musk) + 
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 



deep_HNA <- ggplot(filter(flow_cy, Depth == "Deep"), 
       aes(x = Site, y = HNA_per_uL, fill = Season, shape = Site)) + 
  scale_y_continuous(limits = limits_musk) + ggtitle("Deep samples only") +
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

all_HNA <- ggplot(filter(flow_cy), 
       aes(x = Site, y = HNA_per_uL, fill = Season, shape = Depth)) + ggtitle("All samples") +
  scale_y_continuous(limits = limits_musk) + 
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

HNA_per_uL_plots <- plot_grid(surface_HNA,deep_HNA, all_HNA,
          labels = c("A", "B", "C"), 
          align = "h", nrow = 3, ncol = 1)
HNA_per_uL_plots



# HNA: LNA RATIO
lim_HNA <- c(10,60)

surface_HNA_percent <- ggplot(filter(flow_cy, Depth == "Surface"), 
       aes(x = Site, y = HNA_percent, shape = Site, fill = Season)) + ggtitle("Surface samples only") +
  scale_y_continuous(limits = lim_HNA) + 
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2", color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

deep_HNA_percent <- ggplot(filter(flow_cy, Depth == "Deep"), 
       aes(x = Site, y = HNA_percent, shape = Site, fill = Season)) + ggtitle("Deep samples only") + 
  scale_y_continuous(limits = lim_HNA) + 
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

all_HNA_percent <- ggplot(filter(flow_cy), 
       aes(x = Site, y = HNA_percent, shape = Depth, fill = Season)) + ggtitle("All samples") + 
  scale_y_continuous(limits = lim_HNA) + 
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 


HNA_percent_plots <- plot_grid(surface_HNA_percent,deep_HNA_percent, all_HNA_percent,
          labels = c("A", "B", "C"), 
          align = "h", nrow = 3, ncol = 1)


muskegon_HNA <- plot_grid(HNA_per_uL_plots,HNA_percent_plots,
          labels = c("1", "2"), 
          align = "h", nrow = 1, ncol = 2)

#ggsave("Figures/muskegon_HNA.png", muskegon_HNA, dpi = 600, units = "in", width = 14, height = 10)


ggplot(filter(flow_cy), 
       aes(x = Site, y = cells_per_uL, shape = Depth, fill = Season)) + ggtitle("All samples") + 
  #scale_y_continuous(limits = lim_HNA) + 
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

```


```{r all-HNA-vs-LNA, eval = FALSE}
all_flow_cy <- read.csv2("metadata/flow_cytometry.csv", header = TRUE) %>%
  select(-X) %>%
  # Rename columns so they are more representative of the data within them
  rename(volume_uL = volume, 
         raw_counts = counts,
         HNA_counts = HNA,
         LNA_counts = LNA) %>%
  mutate(cells_per_uL = (raw_counts*dilution)/volume_uL,
         HNA_per_uL = (HNA_counts*dilution)/volume_uL,
         HNA_percent = (HNA_counts)/raw_counts*100,
         LNA_per_uL = (LNA_counts*dilution)/volume_uL,
         LNA_percent = (LNA_counts)/raw_counts*100,
         Nuc_acid_ratio = HNA_per_uL/LNA_per_uL)

all_flow_cy$Season[all_flow_cy$Season == "Winter"] = "Fall"
all_flow_cy$Month[all_flow_cy$Month == "Januari"] = "October"

all_flow_cy$Site <- factor(all_flow_cy$Site,  levels = c("MM110","MM45","MM15","MOT", "MDP", "MBR", "MLB","MIN"))

###  HNA PER uL
lim_all_HNA_per_uL <- c(min(all_flow_cy$HNA_per_uL) - 100,max(all_flow_cy$HNA_per_uL) + 100)


# Set up the main plot
all_Lakes_surface_HNA <- ggplot(filter(all_flow_cy, Depth == "Surface"), 
       aes(x = Site, y = HNA_per_uL, fill = Site, color = Season)) + 
  geom_boxplot(alpha = 0.5, color = "black", show.legend = FALSE) + 
  scale_y_continuous(limits = lim_all_HNA_per_uL) + 
  scale_fill_viridis(option = "viridis", discrete = TRUE) +
  scale_color_brewer(palette="Set1") +
  geom_jitter(size = 3.5) + facet_grid(.~Year, scales = "free_x") + ggtitle("Surface samples only")

all_Lakes_deep_HNA <- ggplot(filter(all_flow_cy, Depth == "Deep"), 
       aes(x = Site, y = HNA_per_uL, fill = Site, color = Season)) + 
  geom_boxplot(alpha = 0.5, color = "black", show.legend = FALSE) + 
  scale_y_continuous(limits = lim_all_HNA_per_uL) + 
  scale_fill_viridis(option = "viridis", discrete = TRUE) +
  scale_color_brewer(palette="Set1") +
  geom_jitter(size = 3.5) + facet_grid(.~Year, scales = "free_x") + ggtitle("Deep samples only")

all_Lakes_all_HNA <- ggplot(filter(all_flow_cy), 
       aes(x = Site, y = HNA_per_uL, fill = Site, color = Season)) + 
  geom_boxplot(alpha = 0.5, color = "black", show.legend = FALSE) +
  scale_y_continuous(limits = lim_all_HNA_per_uL) + 
  scale_fill_viridis(option = "viridis", discrete = TRUE) +
  scale_color_brewer(palette="Set1") +
  geom_jitter(size = 3.5) + facet_grid(.~Year, scales = "free_x") + ggtitle("All samples")


all_Lakes_HNA_per_uL_plots <- plot_grid(all_Lakes_surface_HNA, all_Lakes_deep_HNA, all_Lakes_all_HNA,
          labels = c("A", "B", "C"), 
          align = "h", nrow = 3, ncol = 1)


# HNA: LNA RATIO
lim_all_HNA <- c(min(all_flow_cy$HNA_percent) - 0.05,max(all_flow_cy$HNA_percent) + 0.05)

all_Lakes_surface_HNA_percent <- ggplot(filter(all_flow_cy, Depth == "Surface"), 
       aes(x = Site, y = HNA_percent, fill = Site, color = Season)) + 
  geom_boxplot(alpha = 0.5, color = "black") + 
  scale_y_continuous(limits = lim_all_HNA) + 
  scale_fill_viridis(option = "viridis", discrete = TRUE) +
  scale_color_brewer(palette="Set1") +
  geom_jitter(size = 3.5) + facet_grid(.~Year, scales = "free_x") + ggtitle("Surface samples only")

all_Lakes_deep_HNA_percent <- ggplot(filter(all_flow_cy, Depth == "Deep"), 
       aes(x = Site, y = HNA_percent, fill = Site, color = Season)) + 
  geom_boxplot(alpha = 0.5, color = "black") + 
  scale_y_continuous(limits = lim_all_HNA) + 
  scale_fill_viridis(option = "viridis", discrete = TRUE) +
  scale_color_brewer(palette="Set1") +
  geom_jitter(size = 3.5) + facet_grid(.~Year, scales = "free_x") + ggtitle("Deep samples only")

all_Lakes_all_HNA_percent <- ggplot(filter(all_flow_cy), 
       aes(x = Site, y = HNA_percent, fill = Site, color = Season)) + 
  geom_boxplot(alpha = 0.5, color = "black") +
  scale_y_continuous(limits = lim_all_HNA) + 
  scale_fill_viridis(option = "viridis", discrete = TRUE) +
  scale_color_brewer(palette="Set1") +
  geom_jitter(size = 3.5) + facet_grid(.~Year, scales = "free_x") + ggtitle("All samples")


all_Lakes_HNA_percent_plots <- plot_grid(all_Lakes_surface_HNA_percent,all_Lakes_deep_HNA_percent, all_Lakes_all_HNA_percent,
          labels = c("A", "B", "C"), 
          align = "h", nrow = 3, ncol = 1)



# DO EVERYTHING 
ALLTHEPLOTS <- plot_grid(all_Lakes_HNA_per_uL_plots,all_Lakes_HNA_percent_plots, 
          labels = c("1", "2"), 
          align = "h", nrow = 1, ncol = 2)
ALLTHEPLOTS


#  IS there a relationship between total counts and HNA?
a <- ggplot(all_flow_cy, aes(x = cells_per_uL, y = HNA_per_uL, fill = Site, color = Site, shape = Season)) + 
  geom_point(size = 3.5) + ggtitle("HNA per uL") + 
  scale_color_viridis(option = "viridis", discrete = TRUE)

b <- ggplot(all_flow_cy, aes(x = cells_per_uL, y = HNA_percent, fill = Site, color = Site, shape = Season)) + 
  geom_point(size = 3.5) + ggtitle("Percent HNA")+ 
  scale_color_viridis(option = "viridis", discrete = TRUE)

c <- ggplot(all_flow_cy, aes(x = cells_per_uL, y = LNA_per_uL, fill = Site, color = Site, shape = Season)) + 
  geom_point(size = 3.5) + ggtitle("LNA per uL") + 
  scale_color_viridis(option = "viridis", discrete = TRUE)

d <- ggplot(all_flow_cy, aes(x = cells_per_uL, y = LNA_percent, fill = Site, color = Site, shape = Season)) + 
  geom_point(size = 3.5) + ggtitle("Percent LNA")+ 
  scale_color_viridis(option = "viridis", discrete = TRUE)

e <- plot_grid(a,b,c, d, 
          labels = c("A", "B", "C", "D"), 
          align = "h", nrow = 2, ncol = 2)
e


new <- gather(all_flow_cy, "Nucleic_Acid_Type", "NA_cells_per_uL", LNA_per_uL, HNA_per_uL) %>%
  select(NA_cells_per_uL, Nucleic_Acid_Type, cells_per_uL, Month, Site, Season,Depth,Year)



p1 <- ggplot(new, aes(x = cells_per_uL, y = NA_cells_per_uL, color = Nucleic_Acid_Type)) + 
  geom_point(size = 3, alpha = 0.8) + 
  scale_color_manual(values = c("#675B78", "#EB7B88")) + 
  theme(legend.position = c(0.3, 0.85))

p2 <- ggplot(new, aes(x = cells_per_uL, y = NA_cells_per_uL, color = Season)) + 
  geom_point(size = 3, alpha = 0.8) + 
  scale_color_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) + 
  theme(legend.position = c(0.15, 0.85))


# Ratio of HNA:LNA is not dependent on the number cells per uL!!!!
p3 <- ggplot(all_flow_cy, aes(x = cells_per_uL, y = Nuc_acid_ratio, color = Season)) + 
  geom_point(size = 3, alpha = 0.8) + 
  scale_color_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  geom_smooth(method = "lm") +
  theme(legend.position = c(0.15, 0.85))

p4 <- ggplot(all_flow_cy, aes(x = cells_per_uL, y = Nuc_acid_ratio, color = Season, shape = as.factor(Year))) + 
  geom_point(size = 3, alpha = 0.8) + 
  scale_color_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  geom_smooth(method = "lm") +
  theme(legend.position = c(0.85, 0.75))


ggplot(all_flow_cy, aes(x = cells_per_uL, y = Nuc_acid_ratio, color = as.factor(Year), shape = Season)) + 
  geom_point(size = 5, alpha = 0.8) + 
  scale_color_manual(values = c("#FD7400","#211426", "#BEDB39")) + 
  theme(legend.position = c(0.15, 0.85))


p5 <- plot_grid(p1, p2, p3, p4, 
          labels = c("A", "B", "C", "D"), 
          align = "h", nrow = 2, ncol = 2)

#ggsave("Figures/HNA_LNA.png", p5, dpi = 600, units = "in", width = 10, height = 8)
```





