---
title: "November 2016 Mothur & Oligotyping Analysis of Muskegon 2014-2015"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(eval = TRUE, 
                      echo = TRUE, 
                      cache = TRUE,
                      include = TRUE,
                      collapse = FALSE,
                      dependson = NULL,
                      engine = "R", # Chunks will always have R code, unless noted
                      error = TRUE,
                      fig.path="Figures/cached/",  # Set the figure options
                      fig.align = "center", 
                      fig.width = 7,
                      fig.height = 7)
```

# Load Libraries
```{r load-libraries, echo = FALSE, message = FALSE, include = FALSE}
library(ggplot2)
library(devtools)
library(phyloseq)
library(vegan)
library(tidyr)
library(dplyr)
library(DT)
library(cowplot)
library(wesanderson) # for pretty colors
library(viridis)
library(data.table)
library(stringr)
library(ape)
library(iNEXT) # iNEXT package for alpha diversity measurements 
source("code/Muskegon_functions.R")
source("code/set_colors.R")
```

## Load Mothur OTU Data 
```{r load-data-files}
#  Load in the taxonomy and shared data
otu_tax <- "data/mothur/stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.an.unique_list.0.03.cons.taxonomy"
otu_shared <- "data/mothur/stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.an.unique_list.shared"
otu_data <- import_mothur(mothur_shared_file = otu_shared, mothur_constaxonomy_file = otu_tax)

# Fix the taxonomy names
colnames(tax_table(otu_data)) <- c("Kingdom","Phylum","Class","Order","Family","Genus","Species")

################################################
########## ADD THE PROTEOBACTERIA TO THE PHYLA
phy <- data.frame(tax_table(otu_data))
Phylum <- as.character(phy$Phylum)
Class <- as.character(phy$Class)

for  (i in 1:length(Phylum)){ 
  if (Phylum[i] == "Proteobacteria"){
    Phylum[i] <- Class[i]
  } 
}

phy$Phylum <- Phylum # Add the new phylum level data back to phy
phy$OTU <- row.names(tax_table(otu_data)) # Make a column for OTU

t <- tax_table(as.matrix(phy))

tax_table(otu_data) <- t
################################################

# Sample Names
samp_names <- colnames(otu_table(otu_data))

# Create metadata info
df <- data.frame(matrix(NA, ncol = 1, nrow = length(samp_names)))
colnames(df) <- c("names")
df$names <- samp_names
```



# Load in Metadata to raw phyloseq object
```{r metadata-load}
# Create metadata info
meta_df <- make_muskegon_metadata(df)

# Create a norep_water_name column
meta_df$norep_water_name <- paste(substr(meta_df$names,1,4),substr(meta_df$names,7,9), sep = "")

# Load in the extra metadata for the Muskegon Lake Project
musk_data <- read.csv("data/metadata/processed_muskegon_metadata.csv", header = TRUE) # Load in the extra metada
musk_data_subset <- select(musk_data, -c(lakesite, limnion, month, year, project, season))
complete_meta_df <- left_join(meta_df, musk_data_subset, by = "norep_water_name")
row.names(complete_meta_df) <- complete_meta_df$names

complete_meta_df$water_name <- paste(substr(complete_meta_df$names,1,5),substr(complete_meta_df$names,7,9), sep = "")
complete_meta_df$norep_filter_name <- paste(substr(complete_meta_df$names,1,4),substr(complete_meta_df$names,6,9), sep = "")
```


## Add metadata to the phyloseq objects to Merge samples
```{r merge_samples}
# Add the metadata to our phyloseq object! 
##  Add metadata to OTUs
sample_data(otu_data) <- complete_meta_df
otu_data <- prune_taxa(taxa_sums(otu_data) > 0, otu_data)


# Merged metadata
merged_complete_meta_df<- 
  select(complete_meta_df, -c(names, replicate, nuc_acid_type, water_name)) %>% 
  distinct() %>%
  arrange(norep_filter_name)

## Add Production data from GVSU AWRI provided by the lab of Bopi Biddanda
bopi_data <- read.csv("metadata/production_data.csv") %>%
  dplyr::rename(norep_filter_name = names)  %>% # rename "names" column to "norep_filter_name"
  select(-c(X, limnion, fraction, month, year, season))

## Merge two metadata files together!
df1 <- full_join(merged_complete_meta_df, bopi_data) %>%
  select(-c(Depth, month)) 

# provide row.names to match sample
row.names(df1) <- df1$norep_filter_name

# merge samples for OTUs
otu_merged <- merge_samples(otu_data, group = "norep_filter_name", fun = "sum")
sample_names(otu_merged) == row.names(merged_complete_meta_df) # Sanity check

# Add nice sample information
sample_data(otu_merged) <- df1
```


# Subset Muskegon Samples
```{r subset-muskegon}
# Subset Mukegon Lake samples out of the OTU phyloseq object
otu_merged_musk <- subset_samples(physeq = otu_merged, project == "Muskegon_Lake")
```



# Sample Sequencing Read Counts
```{r seq-read-count}
# Check the sequencing depth of each sample 
sums_otu <- data.frame(rowSums(otu_table(otu_merged_musk)))
colnames(sums_otu) <- "Sample_TotalSeqs"
sums_otu$sample <- row.names(sums_otu)
sums_otu <- arrange(sums_otu, Sample_TotalSeqs) %>%
  filter(Sample_TotalSeqs < 10000)

##  OTU: SUBSAMPLE AT 6600

####  Create a plot of the number of sequences per sample
ggplot(sums_otu, aes(x=reorder(sample, Sample_TotalSeqs), y = Sample_TotalSeqs)) + 
  ylab("Number of Sequences per Sample") +
  geom_bar(stat = "identity", colour="black",fill="cornflowerblue")  + xlab("Sample Name") + 
  ggtitle("OTU: Samples less than 10,000 reads") + 
  theme(axis.text.x = element_text(colour = "black", size=16, angle=45, hjust = 1, vjust = 1))
```


# Phenoflow Alpha Diversity Analysis

## OTU Phenoflow analysis
```{r otu-phenoflow}
# OTU analysis
# Create a new file called "otu_merged_musk_Physeq.RData" that has the phyloseq object
#save(list=c("otu_merged_musk"), file=paste0("Phenoflow/otu_merged_musk_Physeq.RData")) # Will be run on Flux
otu_phenoflow_alpha <- get(load("Phenoflow/otu_phenoflow/Phenoflow_diversity.RData"))

# Add the rownames so that it can be combined with the metadata
otu_phenoflow_alpha <- data.frame(otu_phenoflow_alpha, norep_filter_name = rownames(otu_phenoflow_alpha))
# Create the metadata information 
meta_dat <- data.frame(sample_data(otu_merged_musk))
# Merge phenoflow data and 
otu_phenoflow_alpha <- merge(otu_phenoflow_alpha, meta_dat, by = "norep_filter_name") %>%
  select(-sd.D0.bre, -D0.bre)



### PREPARE DATA FRAMES FOR PHENOFLOW ALPHA DIVERSITY ANALYSIS
free_otu_alpha <- filter(otu_phenoflow_alpha, fraction == "Free" & norep_filter_name != "MOTHJ715" & year == "2015")
part_otu_alpha <- filter(otu_phenoflow_alpha, fraction == "WholePart" & norep_filter_name != "MOTHJ715" & year == "2015")


# Can fractional production be predicted by phenoflow D0 observed richness? 
# FREE LIVING
free_otu_D0_stats <- lm(frac_bacprod ~ D0, data = free_otu_alpha)
free_otu_D0_stats
summary(free_otu_D0_stats)

# PARTICLE
part_otu_D0_stats <- lm(frac_bacprod ~ D0, data = part_otu_alpha)
part_otu_D0_stats
summary(part_otu_D0_stats)

# Plot D0 Observed Richness
otu_pheno_D0 <- ggplot(otu_phenoflow_alpha, aes(x=D0, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = D0 - sd.D0, xmax = D0 + sd.D0), width = 0.2) + 
  ggtitle("OTU: Phenoflow") +
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  scale_x_continuous(limits = c(0,2000)) + 
  scale_y_continuous(limits = c(0,70),expand = c(0,0)) + 
  ylab("Production (μgC/L/hr)") + xlab("Observed Richness (D0)") +
  #geom_smooth(data=subset(otu_phenoflow_alpha, fraction == "Free"), method='lm') + 
  geom_smooth(data=subset(otu_phenoflow_alpha, fraction == "WholePart"), method='lm') + 
  theme(legend.position=c(0.15,0.9), legend.title=element_blank()) +
  annotate("text", x = 1000, y=35, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(free_otu_D0_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(free_otu_D0_stats)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 1600, y=8, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(part_otu_D0_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(part_otu_D0_stats)$coefficients[,4][2]), digits = 4)))


# Can fractional production be predicted by phenoflow chao1? 
# FREE LIVING
free_otu_D0chao_stats <- lm(frac_bacprod ~ D0.chao, data = free_otu_alpha)
free_otu_D0chao_stats
summary(free_otu_D0chao_stats)

# PARTICLE
part_otu_D0chao_stats <- lm(frac_bacprod ~ D0.chao, data = part_otu_alpha)
part_otu_D0chao_stats
summary(part_otu_D0chao_stats)
 
# Plot  chao1
otu_pheno_D0chao <- ggplot(otu_phenoflow_alpha, aes(x=D0.chao, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = D0.chao - sd.D0.chao, xmax = D0.chao + sd.D0.chao), width = 0.2) + 
  ggtitle("OTU: Phenoflow") +
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  scale_x_continuous(limits = c(200,4000)) + 
  scale_y_continuous(limits = c(0,70),expand = c(0,0)) + 
  ylab("Production (μgC/L/hr)") + xlab("Chao1 (D0)") +
  #geom_smooth(data=subset(otu_phenoflow_alpha, fraction == "Free"), method='lm') + 
  geom_smooth(data=subset(otu_phenoflow_alpha, fraction == "WholePart"), method='lm') + 
  theme(legend.position=c(0.15,0.9), legend.title=element_blank()) +
  annotate("text", x = 2000, y=35, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(free_otu_D0chao_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(free_otu_D0chao_stats)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 2700, y=8, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(part_otu_D0chao_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(part_otu_D0chao_stats)$coefficients[,4][2]), digits = 4)))


# Can fractional production be predicted by phenoflow D1 SHANNON ENTROPY? 
# FREE LIVING
free_otu_D1_stats <- lm(frac_bacprod ~ D1, data = free_otu_alpha)
free_otu_D1_stats
summary(free_otu_D1_stats)

# PARTICLE
part_otu_D1_stats <- lm(frac_bacprod ~ D1, data = part_otu_alpha)
part_otu_D1_stats
summary(part_otu_D1_stats)

# Plot D1 SHANNON ENTROPY
otu_pheno_D1 <- ggplot(otu_phenoflow_alpha, aes(x=D1, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = D1 - sd.D1, xmax = D1 + sd.D1), width = 0.2) + 
  ggtitle("OTU: Phenoflow") +
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  scale_x_continuous(limits = c(0,400), expand = c(0,0)) + 
  scale_y_continuous(limits = c(0,70),expand = c(0,0)) + 
  ylab("Production (μgC/L/hr)") + xlab("Shannon Entropy (D1)") +
  #geom_smooth(data=subset(otu_phenoflow_alpha, fraction == "Free"), method='lm') + 
  geom_smooth(data=subset(otu_phenoflow_alpha, fraction == "WholePart"), method='lm') + 
  theme(legend.position=c(0.85,0.9), legend.title=element_blank()) +
  annotate("text", x = 175, y=35, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(free_otu_D1_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(free_otu_D1_stats)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 275, y=7, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(part_otu_D1_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(part_otu_D1_stats)$coefficients[,4][2]), digits = 4)))


# Can fractional production be predicted by phenoflow D2 INVERSE SIMPSON? 
# FREE LIVING
free_otu_D2_stats <- lm(frac_bacprod ~ D2, data = free_otu_alpha)
free_otu_D2_stats
summary(free_otu_D2_stats)

# PARTICLE
part_otu_D2_stats <- lm(frac_bacprod ~ D2, data = part_otu_alpha)
part_otu_D2_stats
summary(part_otu_D2_stats)

# Plot D2 INVERSE SIMPSON
otu_pheno_D2 <- ggplot(otu_phenoflow_alpha, aes(x=D2, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = D2 - sd.D2, xmax = D2 + sd.D2), width = 0.2) + 
  ggtitle("OTU: Phenoflow") +
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  scale_x_continuous(limits = c(0,80), expand = c(0,0)) + 
  scale_y_continuous(limits = c(0,70),expand = c(0,0)) + 
  ylab("Production (μgC/L/hr)") + xlab("Inverse Simpson (D2)") +
  #geom_smooth(data=subset(otu_phenoflow_alpha, fraction == "Free"), method='lm') + 
  geom_smooth(data=subset(otu_phenoflow_alpha, fraction == "WholePart"), method='lm') + 
  theme(legend.position=c(0.85,0.9), legend.title=element_blank()) +
  annotate("text", x = 40, y=45, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(free_otu_D2_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(free_otu_D2_stats)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 50, y=25, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(part_otu_D2_stats)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(part_otu_D2_stats)$coefficients[,4][2]), digits = 4)))



otu_phenoflow <- plot_grid(otu_pheno_D0, otu_pheno_D0chao, otu_pheno_D1, otu_pheno_D2,
          labels = c("A", "B", "C", "D"), 
          align = "h", nrow = 2, ncol = 2)
otu_phenoflow

#ggsave("Figures/Phenoflow_otu.png", otu_phenoflow, dpi = 600, units = "in", width = 10, height = 8)
```



# Vegan Alpha Diversity Analysis
## Remove samples with too few reads
```{r rarefy-and-scale}
##  OTU: SUBSAMPLE AT 6600
# Prune samples out that have too few reads
otu_merged_pruned <- prune_samples(sample_sums(otu_merged) > 6600, otu_merged)
otu_merged_pruned <- prune_taxa(taxa_sums(otu_merged_pruned) > 0, otu_merged_pruned)
min(sample_sums(otu_merged_pruned))

# Scale the samples
scaled_otu_merged_pruned <- otu_merged_pruned %>%
  scale_reads(round = "round") 
```

## Rarefy Read Depth Analysis ran on December 16th, 2016
```{r calculate-alphadiv}
# # Initialize matrices to store richness and evenness estimates
 # nsamp <- nsamples(otu_merged_pruned)
 # min_lib <- min(sample_sums(otu_merged_pruned)) - 1
 # 
 # otu_richness <- matrix(nrow = nsamp, ncol = 100)
 # row.names(otu_richness) <- sample_names(otu_merged_pruned)
 # 
 # otu_evenness <- matrix(nrow = nsamp, ncol = 100)
 # row.names(otu_evenness) <- sample_names(otu_merged_pruned)
 # 
 # otu_shannon <- matrix(nrow = nsamp, ncol = 100)
 # row.names(otu_shannon) <- sample_names(otu_merged_pruned)
 # 
 # # It is always important to set a seed when you subsample so your result is replicable
 # set.seed(777)
 # 
 # for (i in 1:100) {
 #  # Subsample
 #  r <- rarefy_even_depth(otu_merged_pruned, sample.size = min_lib, verbose = FALSE, replace = TRUE)
 # 
 #  # Calculate richness
 #  rich <- as.numeric(as.matrix(estimate_richness(r, measures = "Observed")))
 #  otu_richness[ ,i] <- rich
 # 
 #  # Calculate evenness
 #  even <- as.numeric(as.matrix(estimate_richness(r, measures = "InvSimpson")))
 #  otu_evenness[ ,i] <- even
 # 
 #  # Calculate evenness
 #  shannon <- as.numeric(as.matrix(estimate_richness(r, measures = "Shannon")))
 #  otu_shannon[ ,i] <- shannon
 # 
 # }

#  write.table(otu_evenness, "data/metadata/otu_evenness100_rarefy6664", row.names = TRUE)
#  write.table(otu_richness, "data/metadata/otu_richness100_rarefy6664", row.names = TRUE)
#  write.table(otu_shannon, "data/metadata/otu_shannon100_rarefy6664", row.names = TRUE)
```

## Vegan Alpha Diversity Analysis
### OTU Analysis 
```{r vegan-otu-alphadiv}
# Load values
nsamp <- nsamples(otu_merged_pruned)
min_lib <- min(sample_sums(otu_merged_pruned)) - 1

# Read in the files 
otu_richness <- read.table("metadata/otu_richness100_rarefy6664",  header = TRUE)
otu_evenness <- read.table("metadata/otu_evenness100_rarefy6664", header = TRUE)
otu_shannon <- read.table("metadata/otu_shannon100_rarefy6664", header = TRUE)

# Create a new dataframe to hold the means and standard deviations of richness estimates
norep_filter_name <- row.names(otu_richness)
mean <- apply(otu_richness, 1, mean)
sd <- apply(otu_richness, 1, sd)
measure <- rep("Richness", nsamp)
otu_rich_stats <- data.frame(norep_filter_name, mean, sd, measure)

# Create a new dataframe to hold the means and standard deviations of evenness estimates
norep_filter_name <- row.names(otu_evenness)
mean <- apply(otu_evenness, 1, mean)
sd <- apply(otu_evenness, 1, sd)
measure <- rep("Inverse_Simpson", nsamp)
otu_even_stats <- data.frame(norep_filter_name, mean, sd, measure)

# Create a new dataframe to hold the means and standard deviations of shannon entropy estimates
norep_filter_name <- row.names(otu_shannon)
mean <- apply(otu_shannon, 1, mean)
sd <- apply(otu_shannon, 1, sd)
measure <- rep("Shannon_Entropy", nsamp)
otu_shan_stats <- data.frame(norep_filter_name, mean, sd, measure)

# Create a new dataframe to hold the means and standard deviations of simpsons evenness estimates
norep_filter_name <- row.names(otu_evenness)
mean <- apply(otu_evenness, 1, mean)
sd <- apply(otu_evenness, 1, sd)
measure <- rep("Inverse_Simpson", nsamp)
otu_simpseven_stats <- data.frame(norep_filter_name, mean, sd, measure)

# Calculate Simpson's Evenness into new df called "simps_evenness"
otu_simps_evenness <- inner_join(otu_rich_stats, otu_even_stats, by = "norep_filter_name") %>%
  mutate(mean = mean.y/mean.x,
         sd = sd(mean),
         measure = "Simpsons_Evenness") %>%
  select(norep_filter_name, mean, sd, measure)

# Combine alpha diversity into one dataframe 
otu_alpha <- rbind(otu_rich_stats, otu_even_stats, otu_simps_evenness, otu_shan_stats)
s <- data.frame(sample_data(otu_merged_pruned))
otu_alphadiv <- merge(otu_alpha, s, by = "norep_filter_name") %>%
  filter(project == "Muskegon_Lake")

ggplot(filter(otu_alphadiv, measure == "Richness"), aes(x = norep_filter_name, y = mean, color = lakesite)) +
  geom_point() + facet_grid(project~fraction, scales = "free") + 
  xlab("Sample Name") + ylab("Richness") +
  theme(axis.text.x = element_text(angle = 30))  #Set the x-axis labels)

ggplot(filter(otu_alphadiv, measure == "Inverse_Simpson"), aes(x = norep_filter_name, y = mean, color = lakesite)) +
  geom_point() + facet_grid(project~fraction, scales = "free") + 
  xlab("Sample Name") + ylab("Inverse Simpson") +
  theme(axis.text.x = element_text(angle = 30))  #Set the x-axis labels)

ggplot(filter(otu_alphadiv, measure == "Shannon_Entropy"), aes(x = norep_filter_name, y = mean, color = lakesite)) +
  geom_point() + facet_grid(project~fraction, scales = "free") + 
  xlab("Sample Name") + ylab("Shannon Entropy") +
  theme(axis.text.x = element_text(angle = 30))  #Set the x-axis labels)


###############################
ML_otu_rich_stats <- filter(otu_alphadiv, measure == "Richness" & project == "Muskegon_Lake" & fraction %in% c("WholePart", "Free") & year == "2015")

# Can fractional production be predicted by richness? 
free_ML_otu_rich_stats <- filter(ML_otu_rich_stats, fraction == "Free")
freeprod_ML_otu_rich <- lm(frac_bacprod ~ mean, data = free_ML_otu_rich_stats)
freeprod_ML_otu_rich
summary(freeprod_ML_otu_rich)

part_ML_abs_rich_stats <- filter(ML_otu_rich_stats, fraction == "WholePart")
partprod_MLotu_rich <- lm(frac_bacprod ~ mean, data = part_ML_abs_rich_stats)
partprod_MLotu_rich
summary(partprod_MLotu_rich)

# Plot 
otu_rich_vegan <- ggplot(ML_otu_rich_stats, aes(x=mean, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = mean - sd, xmax = mean + sd), width = 0.2) + 
  ggtitle("OTU: Vegan") +
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  scale_x_continuous(expand = c(0,0), limits = c(0,1250)) + 
  scale_y_continuous(limits = c(0,70),expand = c(0,0)) + 
  ylab("Production (μgC/L/hr)") + xlab("Observed Richness (D0)") +
  #geom_smooth(data=subset(ML_otu_rich_stats, fraction == "Free"), method='lm') + 
  geom_smooth(data=subset(ML_otu_rich_stats, fraction == "WholePart"), method='lm') + 
  theme(legend.position=c(0.15,0.9),        
        legend.title=element_blank()) +
  annotate("text", x = 800, y=35, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(freeprod_ML_otu_rich)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(freeprod_ML_otu_rich)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 900, y=5, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(partprod_MLotu_rich)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(partprod_MLotu_rich)$coefficients[,4][2]), digits = 4)));



######################################################### SHANNON ENTROPY
#  Subset a dataframe with the key information
ML_otu_shannon_stats <- filter(otu_alphadiv, 
                                   measure == "Shannon_Entropy" & 
                                     project == "Muskegon_Lake" & 
                                     fraction %in% c("WholePart", "Free") & 
                                     year == "2015")

# Can fractional production be predicted by simpsevenness? 
free_ML_otu_shannon_stats <- filter(ML_otu_shannon_stats, fraction == "Free")
freeprod_ML_otu_shannon <- lm(frac_bacprod ~ mean, data = free_ML_otu_shannon_stats)
freeprod_ML_otu_shannon
summary(freeprod_ML_otu_shannon)

part_ML_abs_shannon_stats <- filter(ML_otu_shannon_stats, fraction == "WholePart")
partprod_MLotu_shannon <- lm(frac_bacprod ~ mean, data = part_ML_abs_shannon_stats)
partprod_MLotu_shannon
summary(partprod_MLotu_shannon)

# Plot 
otu_shannon_vegan <- ggplot(ML_otu_shannon_stats, aes(x=mean, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = mean - sd, xmax = mean + sd), width = 0.2) + 
  ggtitle("OTU: Vegan") +
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  scale_x_continuous(expand = c(0,0), limits=c(3, 6)) + 
  scale_y_continuous(limits = c(0,70),expand = c(0,0)) + 
  ylab("Production (μgC/L/hr)") + xlab("Shannon Entropy (D1)") +
  #geom_smooth(data=subset(ML_otu_shannon_stats, fraction == "Free"), method='lm') + 
  geom_smooth(data=subset(ML_otu_shannon_stats, fraction == "WholePart"), method='lm') + 
  theme(legend.position=c(0.15,0.9),        
        legend.title=element_blank()) +
  annotate("text", x = 3.5, y=35, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(freeprod_ML_otu_shannon)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(freeprod_ML_otu_shannon)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 5.35, y=5, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(partprod_MLotu_shannon)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(partprod_MLotu_shannon)$coefficients[,4][2]), digits = 4))); 





######################################################### INVERSE SIMPSON
#  Subset a dataframe with the key information
ML_otu_invsimps_stats <- filter(otu_alphadiv, 
                                   measure == "Inverse_Simpson" & 
                                     project == "Muskegon_Lake" & 
                                     fraction %in% c("WholePart", "Free") & 
                                     year == "2015")

# Can fractional production be predicted by invsimpsness? 
free_ML_otu_invsimps_stats <- filter(ML_otu_invsimps_stats, fraction == "Free")
freeprod_ML_otu_invsimps <- lm(frac_bacprod ~ mean, data = free_ML_otu_invsimps_stats)
freeprod_ML_otu_invsimps
summary(freeprod_ML_otu_invsimps)

part_ML_abs_invsimps_stats <- filter(ML_otu_invsimps_stats, fraction == "WholePart")
partprod_MLotu_invsimps <- lm(frac_bacprod ~ mean, data = part_ML_abs_invsimps_stats)
partprod_MLotu_invsimps
summary(partprod_MLotu_invsimps)

# Plot Simpson's Evenness
otu_invsimps_vegan <- ggplot(ML_otu_invsimps_stats, aes(x=mean, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = mean - sd, xmax = mean + sd), width = 0.2) + 
  ggtitle("OTU: Vegan") +
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  scale_x_continuous(limits = c(0,100), expand = c(0,0)) + 
  scale_y_continuous(limits = c(0,70),expand = c(0,0)) + 
  ylab("Production (μgC/L/hr)") + xlab("Inverse Simpson (D2)") +
  #geom_smooth(data=subset(ML_otu_invsimps_stats, fraction == "Free"), method='lm') + 
  geom_smooth(data=subset(ML_otu_invsimps_stats, fraction == "WholePart"), method='lm') + 
  theme(legend.position=c(0.85,0.9),        
        legend.title=element_blank()) +
  annotate("text", x = 58, y=35, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(freeprod_ML_otu_invsimps)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(freeprod_ML_otu_invsimps)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 63, y=5, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(partprod_MLotu_invsimps)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(partprod_MLotu_invsimps)$coefficients[,4][2]), digits = 4))); 




######################################################### SIMPSON'S EVENNESS
#  Subset a dataframe with the key information
ML_otu_simpseven_stats <- filter(otu_alphadiv, 
                                   measure == "Simpsons_Evenness" & 
                                     project == "Muskegon_Lake" & 
                                     fraction %in% c("WholePart", "Free") & 
                                     year == "2015")

# Can fractional production be predicted by simpsevenness? 
free_ML_otu_simpseven_stats <- filter(ML_otu_simpseven_stats, fraction == "Free")
freeprod_ML_otu_simpseven <- lm(frac_bacprod ~ mean, data = free_ML_otu_simpseven_stats)
freeprod_ML_otu_simpseven
summary(freeprod_ML_otu_simpseven)

part_ML_abs_simpseven_stats <- filter(ML_otu_simpseven_stats, fraction == "WholePart")
partprod_MLotu_simpseven <- lm(frac_bacprod ~ mean, data = part_ML_abs_simpseven_stats)
partprod_MLotu_simpseven
summary(partprod_MLotu_simpseven)

# Plot 
otu_simpseven_vegan <- ggplot(ML_otu_simpseven_stats, aes(x=mean, y=frac_bacprod, color = fraction)) + 
  geom_point(size = 3.5) +
  ggtitle("OTU: Vegan") +
  scale_color_manual(values = c("firebrick3","cornflowerblue"), limits = c("WholePart", "Free")) +
  scale_x_continuous(expand = c(0,0), limits=c(0, 0.09)) + 
  scale_y_continuous(limits = c(0,70),expand = c(0,0)) + 
  ylab("Production (μgC/L/hr)") + xlab("Simpson's Evenness") +
  #geom_smooth(data=subset(ML_otu_simpseven_stats, fraction == "Free"), method='lm') + 
  geom_smooth(data=subset(ML_otu_simpseven_stats, fraction == "WholePart"), method='lm') + 
  theme(legend.position=c(0.15,0.9),        
        legend.title=element_blank()) +
  annotate("text", x = 0.03, y=35, color = "cornflowerblue", fontface = "bold",
           label = paste("R2 =", round(summary(freeprod_ML_otu_simpseven)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(freeprod_ML_otu_simpseven)$coefficients[,4][2]), digits = 4))) + 
  annotate("text", x = 0.015, y=7, color = "firebrick3", fontface = "bold",
           label = paste("R2 =", round(summary(partprod_MLotu_simpseven)$adj.r.squared, digits = 4), "\n", 
                         "p =", round(unname(summary(partprod_MLotu_simpseven)$coefficients[,4][2]), digits = 4))); 

otu_vegan <- plot_grid(otu_rich_vegan, otu_simpseven_vegan, otu_shannon_vegan, otu_invsimps_vegan,
          labels = c("A", "B", "C", "D"), 
          align = "h", nrow = 2, ncol = 2)
otu_vegan

ggsave("Figures/vegan_otu_alpha_vs_prod.png", otu_vegan, dpi = 600, units = "in", width = 10, height = 8)
```




#  Stacked Bar plots for taxonomic structure
## OTU  Phylum Stacked Bar 
```{r otu-stacked-bar}
musk_scaled_otu_merged_pruned <- subset_samples(scaled_otu_merged_pruned, project == "Muskegon_Lake" & 
                                           year == "2015" &
                                           limnion == "Top" &
                                           fraction %in% c("WholePart", "Free"))

# Fix month levels in sample_data
sample_data(musk_scaled_otu_merged_pruned)$fraction <- factor(
  sample_data(musk_scaled_otu_merged_pruned)$fraction, 
  levels = c("WholePart", "Free"))

sample_data(musk_scaled_otu_merged_pruned)$lakesite <- factor(
  sample_data(musk_scaled_otu_merged_pruned)$lakesite, 
  levels = c("MOT", "MDP", "MBR", "MIN"))



musk_scaled_otu_merged_pruned_phylum <- musk_scaled_otu_merged_pruned %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa
  arrange(Phylum)                                      # Sort data frame alphabetically by phylum


# Set colors for plotting
phylum_colors <- c(
  "#CBD588", "#5F7FC7", "orange","#DA5724", "#508578", "plum1",
   "firebrick4", "white", "#D14285","seagreen3", "gold", 
  "#8569D5", "firebrick1", "peachpuff"
)


# Plot 
otu_stack_bar <- ggplot(musk_scaled_otu_merged_pruned_phylum, aes(x = season, y = Abundance, fill = Phylum)) + 
  facet_grid(fraction~lakesite) +
  geom_bar(stat = "identity", color = "black") +
  scale_fill_manual(values = phylum_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  ggtitle("Phylum Composition of 2015 Muskegon Lake \n Bacterial Communities by Sampling Site") +
  theme(strip.background = element_rect(fill = "white"),
        axis.text.x = element_text(colour = "black", angle=45, hjust = 1, vjust = 1))

#ggsave("Figures/otu_phylum_stackedbar.png", otu_stack_bar, dpi = 600, units = "in", width = 12, height = 8)

```




# Environmental Conditions 
```{r}
prod_data <- data.frame(sample_data(musk_scaled_otu_merged_pruned))

ggplot(prod_data, aes(x = season, y = tot_bacprod, color = lakesite, group = lakesite, linetype = lakesite)) + 
  geom_point(size = 3.5) +   geom_line(size = 2) +
  ylab("Total Bacterial\n Production") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1))

#### Plot many environ variables
tot_prod_plot <- ggplot(prod_data, aes(x = season, y = tot_bacprod, color = lakesite, group = lakesite)) + 
  facet_grid(.~lakesite) +
  geom_point(size = 3.5) + geom_line() +
  ylab("Total Bacterial\n Production") +
  theme(strip.background = element_rect(fill = "white"),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1),
        legend.position="none")

chla_plot <- ggplot(prod_data, aes(x = season, y = Chl_Lab_ugperL, color = lakesite, group = lakesite)) + 
  facet_grid(.~lakesite) +
  geom_point(size = 3.5) + geom_line() +
  ylab("Chlorophyll a") +
  theme(strip.background = element_rect(fill = "white"),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1),
        legend.position="none")

ph_plot <- ggplot(prod_data, aes(x = season, y = pH, color = lakesite, group = lakesite)) + 
  facet_grid(.~lakesite) +
  geom_point(size = 3.5) + geom_line() +
  ylab("pH") +
  theme(strip.background = element_rect(fill = "white"),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1),
        legend.position="none")

tp_plot <- ggplot(prod_data, aes(x = season, y = TP_ugperL, color = lakesite, group = lakesite)) + 
  facet_grid(.~lakesite) +
  geom_point(size = 3.5) + geom_line() +
  ylab("TP_ugperL") +
  theme(strip.background = element_rect(fill = "white"),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1),
        legend.position="none")

no3_plot <- ggplot(prod_data, aes(x = season, y = NO3_mgperL, color = lakesite, group = lakesite)) + 
  facet_grid(.~lakesite) +
  geom_point(size = 3.5) + geom_line() +
  ylab("NO3_mgperL") +
  theme(strip.background = element_rect(fill = "white"),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1),
        legend.position="none")

nh3_plot <- ggplot(prod_data, aes(x = season, y = NH3_mgperL, color = lakesite, group = lakesite)) + 
  facet_grid(.~lakesite) +
  geom_point(size = 3.5) + geom_line() +
  ylab("NH3_mgperL") +
  theme(strip.background = element_rect(fill = "white"),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black", angle=15, hjust = 1, vjust = 1),
        legend.position="none")

environ_seasons <- plot_grid(tot_prod_plot, chla_plot, ph_plot, tp_plot, no3_plot, nh3_plot,
          labels = c("A", "B", "C", "D", "E", "F"), 
          align = "h", nrow = 6, ncol = 1)
environ_seasons

#ggsave("Figures/environ_seasons.png", environ_seasons, dpi = 600, units = "in", width = 12, height = 12)
```



# Phylum production analysis
```{r phylum-div-prod, eval = FALSE}
# Michelle Berry's subsetting function, similar to phyloseq::subset_taxa, except taxa can 
# be passed as arguments within functions without weird environment errors
#
# Args:
#   physeq: a phyloseq object
#   taxrank: taxonomic rank to filter on
#   taxa: a vector of taxa groups to filter on
#
# Returns: 
#   a phyloseq object subsetted to the x taxa in taxrank
my_subset_taxa <- function(physeq, taxrank, taxa) {
  physeq_tax_sub <- tax_table(physeq)[tax_table(physeq)[ , taxrank] %in% taxa, ]
  tax_table(physeq) <- physeq_tax_sub
  return(physeq)
}

free_samps <- subset_samples(musk_scaled_otu_merged_pruned, fraction == "Free")


# Initialize parameters
trials <- 100
min_lib <- min(sample_sums(musk_scaled_otu_merged_pruned)) # Depth we are rarefying to

# Groups to estimate alpha diversity for 
mytaxa <- c("Acidobacteria", "Actinobacteria", "Alphaproteobacteria", "Armatimonadetes",
            "Bacteroidetes", "Betaproteobacteria", "Chlorobi", "Chloroflexi",
            "Cyanobacteria","Deltaproteobacteria", "Gammaproteobacteria", 
            "Planctomycetes","Verrucomicrobia")
names(mytaxa) <- mytaxa

# Taxonomic ranks of mytaxa
mytaxa_taxrank <- c("Phylum", "Phylum", "Phylum", "Phylum", "Phylum", "Phylum", 
                      "Phylum", "Phylum", "Phylum", "Phylum", "Phylum", "Phylum", "Phylum")
names(mytaxa_taxrank) <- mytaxa

# Data frame to hold alpha diversity estimates over trials
alphadiv_df <- data.frame(matrix(nrow = nsamples(musk_scaled_otu_merged_pruned), ncol = trials))

# Initialize empty df's for richness and evenness of all taxa in mytaxa
richness <- lapply(mytaxa, function(x) {return(alphadiv_df)} )
shannon <- lapply(mytaxa, function(x) {return(alphadiv_df)} )
invsimps <- lapply(mytaxa, function(x) {return(alphadiv_df)} )
simpseven <- lapply(mytaxa, function(x) {return(alphadiv_df)} )

alphadiv_list <- list(richness = richness, shannon = shannon, invsimps = invsimps, simpseven = simpseven)


# Set the seed for reproducibility
set.seed(3)

# Run trials to subsample and estimate diversity
for (i in 1:100) {
  
  # Subsample
  rarefied_physeq <- rarefy_even_depth(musk_scaled_otu_merged_pruned, sample.size = min_lib, verbose = FALSE, replace = TRUE)
  
  # Generate alpha-diversity estimates for each taxonomic group
  for (t in mytaxa) {

    # Subset the taxa
    physeq_sub <- my_subset_taxa(physeq = rarefied_physeq, 
                                 taxrank = mytaxa_taxrank[t], 
                                 taxa = t)
    
    # Calculate observed richness for that group and store value in a df column
    richness <- estimate_richness(physeq_sub, measures = "Observed")[ ,1]
    alphadiv_list$richness[[t]][ ,i] <- richness
    
    # Calculate observed shannon for that group and store value in a df column
    shannon <- estimate_richness(physeq_sub, measures = "Shannon")[ ,1]
    alphadiv_list$shannon[[t]][ ,i] <- shannon
    
    # Calculate observed invsimps for that group and store value in a df column
    invsimps <- estimate_richness(physeq_sub, measures = "InvSimpson")[ ,1]
    alphadiv_list$invsimps[[t]][ ,i] <- invsimps
     
    # Calculate Simpson's Evenness for that group and store value in a df column
    alphadiv_list$simpseven[[t]][ ,i] <- (estimate_richness(physeq_sub, measures = "InvSimpson")[ ,1]/richness)

  }
}

# Calculate the means of richness and inverse simpson from the 100 trials
alphadiv_est <- lapply(alphadiv_list, function(div_measure) {
    lapply(div_measure, function(taxa_group) {
        alpha_mean <- rowMeans(taxa_group)
        return(alpha_mean)
    })  
})

# Convert alphadiv_est richness and simpson's E lists into wide data frames
l <- lapply(alphadiv_est, function(x) {
  # convert from list to data.frame
  est_df <- plyr::ldply(.data = x, .fun = data.frame)
  names(est_df) <- c("Taxa", "Diversity")
  
  # Add in SampleID column and spread to wide format
  r <- est_df %>%
    mutate(norep_filter_name = rep(sample_names(musk_scaled_otu_merged_pruned), length(mytaxa)))
  return(r)
})

# Merge sample metadata with these estimates
merge_dat <- data.frame(sample_data(musk_scaled_otu_merged_pruned)) %>%
  select(norep_filter_name, Chl_Lab_ugperL, TP_ugperL, pH, frac_bacprod, NH3_mgperL, NO3_mgperL,
         lakesite, fraction, season) #%>%
  #mutate(logChla = log(Chla)) %>%
  #mutate(logPhyco = log(Phycocyanin + 0.1)) %>%
  #mutate(logTP = log(TP)) %>%
  #mutate(logTurb = log(Turbidity))

# Create a df with a "Diversity" column that includes richness and inv. simpson,
# and log-chl a values from erie sample_data
alpha_comb1 <- l$richness %>% 
  left_join(y = l$shannon, by = c("Taxa", "norep_filter_name")) %>%   # Join the richness and inv_simp df's
  rename(Richness = Diversity.x, Shannon = Diversity.y) 

alpha_comb2 <- l$invsimps %>%   
  left_join(y = l$simpseven, by = c("Taxa", "norep_filter_name")) %>%   # Join the richness and inv_simp df's
  rename(Inverse_Simpson = Diversity.x, Simpsons_Evenness = Diversity.y) 

alpha_comb_final <- alpha_comb1 %>%
  left_join(y = alpha_comb2, by = c("Taxa", "norep_filter_name")) %>%
  left_join(merge_dat, by = "norep_filter_name") %>%                  # Join with merged nutrient data
  gather(key = "Alphadiv", value = "Estimate", Richness, Shannon, Inverse_Simpson, Simpsons_Evenness)
```


# High vs Low nucleic acid bacteria
## Muskegon Lake Only
```{r muskegon-HNA-vs-LNA}
flow_cy <- read.csv2("metadata/flow_cytometry.csv", header = TRUE) %>%
  select(-X) %>%
  filter(Lake == "Muskegon") %>%
  # Rename columns so they are more representative of the data within them
  rename(volume_uL = volume, 
         raw_counts = counts,
         HNA_counts = HNA,
         LNA_counts = LNA) %>%
  mutate(cells_per_uL = (raw_counts*dilution)/volume_uL,
         HNA_per_uL = (HNA_counts*dilution)/volume_uL,
         HNA_percent = (HNA_counts)/raw_counts*100,
         LNA_per_uL = (LNA_counts*dilution)/volume_uL,
         LNA_percent = (LNA_counts)/raw_counts,
         Nuc_acid_ratio = HNA_per_uL/LNA_per_uL)

# Fix the factor levels for nice plotting
flow_cy$Site <- factor(flow_cy$Site,  levels = c("MOT", "MDP", "MBR", "MIN"))
flow_cy$Season <- factor(flow_cy$Season,  levels = c("Spring", "Summer", "Fall"))

# Create a new column so we can merge with other data frames
flow_cy$norep_water_name <- paste(substring(flow_cy$Sample_16S,1,4), substring(flow_cy$Sample_16S,7,9), sep = "")

# Prepare the dataframes with Bopi's data
free_sample_physeq <- subset_samples(otu_merged_musk, fraction == "Free")
free_samples <-  sample_data(free_sample_physeq)



# Read in the Flow Cytometry Diversity data that Ruben calculated with Phenoflow 
flow_div <- read.csv("Phenoflow/Diversity.fcm.marian.csv", header = TRUE) %>%
  select(-X) 

flow_data <- left_join(flow_div, flow_cy, by = "Sample") %>%
    filter(Lake == "Muskegon") 

HNA_prod <- left_join(flow_data, free_samples, by = "norep_water_name") 

##  PREPARE A PHYLOSEQ OBJECT FOR 
HNA_prod2 <- left_join(free_samples, flow_data, by = "norep_water_name") 
row.names(HNA_prod2) <- HNA_prod2$norep_filter_name
sample_data(free_sample_physeq) <- HNA_prod2

```


# Calculate the Absolute abundances!
```{r prepare-peter-physeq}
### MODIFY TO ABSOLUTE ABUNDANCES!

# First let's calculate the proportion
peter_physeq <- free_sample_physeq 
rel_abund_physeq <- peter_physeq %>%
  transform_sample_counts(function(x) {x/sum(x)} ) # Transform to rel. abundance

# Multiply the proportion by the total number of cells
relabund_otu <- otu_table(rel_abund_physeq)
absabund_counts_otu <- relabund_otu * sample_data(peter_physeq)$cells_per_uL
stopifnot(rowSums(absabund_counts_otu) == sample_data(peter_physeq)$cells_per_uL) # Sanity check
otu_table(peter_physeq) <- absabund_counts_otu # Reassign the otu_table to be the absolute counts
otu_table(peter_physeq)[1:5,1:15]

## From counts to absolute abundance proprtions 
absabund_proportion_otu <- peter_physeq %>%
  transform_sample_counts(function(x) {x/sum(x)} ) # Transform to rel. abundance
otu_table(absabund_proportion_otu)[1:5,1:15] == otu_table(rel_abund_physeq)[1:5,1:15] # Relative and absolute proportions are the same!

# Final otu data frames
peter_otu_counts <- absabund_counts_otu # OTU COUNTS TABLE
peter_otu_proportion <- otu_table(absabund_proportion_otu) # OTU PROPORTION TABLE

# Write the files 
#write.table(peter_otu_counts, "peter/muskegon_absolute_otu_counts.tsv", sep = "\t")
#write.table(peter_otu_proportion, "peter/muskegon_otu_proportions.tsv", sep = "\t")

#write.xlsx(x = peter_otu_counts, file = "peter/muskegon_absolute_otu_counts.xlsx")
#write.xlsx(x = peter_otu_proportion, file = "peter/muskegon_otu_proportions.xlsx")

HNA_prod3 <- HNA_prod2 %>%
  select(norep_filter_name, lakesite, limnion, year, season,
         tot_bacprod,
         cells_per_uL, HNA_per_uL, HNA_percent, LNA_per_uL, LNA_percent) %>%
  rename(depth = limnion)

row.names(HNA_prod3) = NULL

#write.table(HNA_prod3, "peter/sample_data.tsv", sep = "\t")
#write.xlsx(x = HNA_prod3, file = "peter/sample_data.xlsx")


tot_bacprod_plot1 <- ggplot(HNA_prod3, aes(y = tot_bacprod, x = lakesite, color = lakesite, fill = lakesite)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.9)) +
  scale_color_manual(values = lakesite_colors) +
  scale_fill_manual(values = lakesite_colors)


tot_bacprod_plot2 <- ggplot(HNA_prod3, aes(y = tot_bacprod, x = lakesite, color = year, fill = year)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.9)) +
  scale_color_manual(values = year_colors) +
  scale_fill_manual(values = year_colors)


tot_bacprod_plot3 <- ggplot(HNA_prod3, aes(y = tot_bacprod, x = lakesite, color = season, fill = season)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.9)) +
  scale_color_manual(values = season_colors) +
  scale_fill_manual(values = season_colors)

a_plot1 <- ggplot(HNA_prod3, aes(y = HNA_percent, x = lakesite, color = lakesite, fill = lakesite)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.9)) +
  scale_color_manual(values = lakesite_colors) +
  scale_fill_manual(values = lakesite_colors)


a_plot2 <- ggplot(HNA_prod3, aes(y = HNA_percent, x = lakesite, color = year, fill = year)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.9)) +
  scale_color_manual(values = year_colors) +
  scale_fill_manual(values = year_colors)


a_plot3 <- ggplot(HNA_prod3, aes(y = HNA_percent, x = lakesite, color = season, fill = season)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.9)) +
  scale_color_manual(values = season_colors) +
  scale_fill_manual(values = season_colors)


tot_bacprod_plotz <- plot_grid(tot_bacprod_plot1, tot_bacprod_plot2, tot_bacprod_plot3, 
           labels = c("A", "B", "C"), 
           align = "h", nrow = 3, ncol = 1)

#ggsave("Figures/tot_bacprod_factors.png", tot_bacprod_plotz, dpi = 600, units = "in", width = 5, height = 6)


# Write a season table for peter
season_df <- HNA_prod3 %>%
  select(norep_filter_name, season)
write.table(season_df, "peter/sample_season_data.tsv", sep = "\t", row.names = FALSE)



a_plotz <- plot_grid(tot_bacprod_plot1, a_plot1,tot_bacprod_plot2, a_plot2, tot_bacprod_plot3, a_plot3,
          labels = c("A", "B", "C", "D", "E", "F"), 
          align = "h", nrow = 3, ncol = 2)


ggplot(HNA_prod3, aes(y = HNA_percent, x = lakesite, color = lakesite, fill = lakesite)) +
  geom_boxplot(alpha = 0.5) + 
  geom_point(size = 3, position=position_jitterdodge(dodge.width=0.6)) +
  scale_color_manual(values = lakesite_colors) +
  scale_fill_manual(values = lakesite_colors)


```


### HNA VS LNA 
```{r}
####  PRODUCTION VS HNA

ggplot(HNA_prod, aes(x = Site, y = tot_bacprod, fill = Season, shape = Site)) + ggtitle("Surface samples only") +
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 


r1 <- ggplot(HNA_prod, aes(x = HNA_per_uL, y = tot_bacprod)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site), show.legend = FALSE) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

r2 <- ggplot(HNA_prod, aes(x = HNA_percent, y = tot_bacprod)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site), show.legend = FALSE) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

r3 <- ggplot(HNA_prod, aes(x = cells_per_uL, y = tot_bacprod)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site)) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

r4 <-ggplot(HNA_prod, aes(x = Nuc_acid_ratio, y = tot_bacprod)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site)) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

r5 <- ggplot(HNA_prod, aes(x = LNA_per_uL, y = tot_bacprod)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site), show.legend = FALSE) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

r6 <- ggplot(HNA_prod, aes(x = LNA_percent, y = tot_bacprod)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site), show.legend = FALSE) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 


rr <- plot_grid(r1, r5, r3, r2, r6, r4, 
          labels = c("A", "B", "C", "D", "E", "F"), 
          align = "h", nrow = 2, ncol = 3)

#ggsave("Figures/HNA_LNA_vs_production.png", rr, dpi = 600, units = "in", width = 14, height = 8)



##################################################################################################################################
##################################################################################################################################
##################################################################################################################################


a <- ggplot(HNA_prod, aes(x = HNA_per_uL, y = attached_bac)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site)) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) +
  theme(legend.position = c(0.15, 0.75))

b <- ggplot(HNA_prod, aes(x = HNA_per_uL, y = perc_attached_abund)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site)) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) +
  theme(legend.position = c(0.15, 0.75))

ab <- plot_grid(a, b,
          labels = c("A", "B"), 
          align = "h", nrow = 1, ncol = 2)

#ggsave("Figures/HNA_vs_attached_bac.png", ab, dpi = 600, units = "in", width = 10, height = 6)




ggplot(HNA_prod, aes(x = cells_per_uL, y = total_bac_abund/1000)) + ggtitle("Surface samples only") +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5, aes(fill = Season, shape = Site)) + #facet_grid(.~Year) + 
  geom_smooth(method = "lm", color = "black") + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) +
  theme(legend.position = c(0.2, 0.75))


##################################################################################################################################
##################################################################################################################################
##################################################################################################################################
# Plot D0 Observed Richness
d0 <- ggplot(HNA_prod, aes(x=D0, y=tot_bacprod)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = D0 - D0.sd, xmax = D0 + D0.sd), width = 0.2) + 
  ggtitle("Phenotypic Div: Phenoflow") +
  ylab("Total Production (μgC/L/hr)") + xlab("Observed Richness (D0)") +
  theme(legend.position=c(0.15,0.9), legend.title=element_blank()) +
  geom_smooth(method = "lm", color = "black") + 
  scale_x_continuous(limits = c(13800, 16300)) 


d1 <- ggplot(HNA_prod, aes(x=D1, y=tot_bacprod)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = D1 - D1.sd, xmax = D1 + D1.sd), width = 0.2) + 
  ggtitle("Phenotypic Div: Phenoflow") +
  ylab("Total Production (μgC/L/hr)") + xlab("Shannon Entropy (D1)") +
  theme(legend.position=c(0.15,0.9), legend.title=element_blank()) +
  geom_smooth(method = "lm", color = "black") +
  scale_x_continuous(limits = c(3000, 3700)) 

d2 <- ggplot(HNA_prod, aes(x=D2, y=tot_bacprod)) + 
  geom_point(size = 3.5) + geom_errorbarh(aes(xmin = D2 - D2.sd, xmax = D2 + D2.sd), width = 0.2) + 
  ggtitle("Phenotypic Div: Phenoflow") +
  ylab("Total Production (μgC/L/hr)") + xlab("Inverse Simpson (D2)") +
  theme(legend.position=c(0.15,0.9), legend.title=element_blank()) +
  geom_smooth(method = "lm", color = "black") + 
  scale_x_continuous(limits = c(1850, 2300)) 


dd <- plot_grid(d0, d1, d2,
          labels = c("A", "B", "C"), 
          align = "h", nrow = 1, ncol = 3)

ggsave("Figures/phenotypic_div_vs_total_production.png", dd, dpi = 600, units = "in", width = 10, height = 6)

##################################################################################################################################
##################################################################################################################################
##################################################################################################################################
###  HNA PER uL
limits_musk <- c(600,3700)

surface_HNA <- ggplot(filter(flow_cy, Depth == "Surface"), 
       aes(x = Site, y = HNA_per_uL, fill = Season, shape = Site)) + ggtitle("Surface samples only") +
  scale_y_continuous(limits = limits_musk) + 
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 



deep_HNA <- ggplot(filter(flow_cy, Depth == "Deep"), 
       aes(x = Site, y = HNA_per_uL, fill = Season, shape = Site)) + 
  scale_y_continuous(limits = limits_musk) + ggtitle("Deep samples only") +
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

all_HNA <- ggplot(filter(flow_cy), 
       aes(x = Site, y = HNA_per_uL, fill = Season, shape = Depth)) + ggtitle("All samples") +
  scale_y_continuous(limits = limits_musk) + 
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

HNA_per_uL_plots <- plot_grid(surface_HNA,deep_HNA, all_HNA,
          labels = c("A", "B", "C"), 
          align = "h", nrow = 3, ncol = 1)
HNA_per_uL_plots



# HNA: LNA RATIO
lim_HNA <- c(10,60)

surface_HNA_percent <- ggplot(filter(flow_cy, Depth == "Surface"), 
       aes(x = Site, y = HNA_percent, shape = Site, fill = Season)) + ggtitle("Surface samples only") +
  scale_y_continuous(limits = lim_HNA) + 
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2", color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

deep_HNA_percent <- ggplot(filter(flow_cy, Depth == "Deep"), 
       aes(x = Site, y = HNA_percent, shape = Site, fill = Season)) + ggtitle("Deep samples only") + 
  scale_y_continuous(limits = lim_HNA) + 
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

all_HNA_percent <- ggplot(filter(flow_cy), 
       aes(x = Site, y = HNA_percent, shape = Depth, fill = Season)) + ggtitle("All samples") + 
  scale_y_continuous(limits = lim_HNA) + 
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 


HNA_percent_plots <- plot_grid(surface_HNA_percent,deep_HNA_percent, all_HNA_percent,
          labels = c("A", "B", "C"), 
          align = "h", nrow = 3, ncol = 1)


muskegon_HNA <- plot_grid(HNA_per_uL_plots,HNA_percent_plots,
          labels = c("1", "2"), 
          align = "h", nrow = 1, ncol = 2)

#ggsave("Figures/muskegon_HNA.png", muskegon_HNA, dpi = 600, units = "in", width = 14, height = 10)


ggplot(filter(flow_cy), 
       aes(x = Site, y = cells_per_uL, shape = Depth, fill = Season)) + ggtitle("All samples") + 
  #scale_y_continuous(limits = lim_HNA) + 
  geom_boxplot(alpha = 0.5, fill = "#C5DBF2",color = "black", show.legend = FALSE) +
  scale_fill_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  scale_shape_manual(values = c(21, 22, 23, 25)) +
  geom_point(size = 3.5) + facet_grid(.~Year) + 
  guides(fill = guide_legend(override.aes = list(shape = 24))) 

```


```{r all-HNA-vs-LNA}
all_flow_cy <- read.csv2("metadata/flow_cytometry.csv", header = TRUE) %>%
  select(-X) %>%
  # Rename columns so they are more representative of the data within them
  rename(volume_uL = volume, 
         raw_counts = counts,
         HNA_counts = HNA,
         LNA_counts = LNA) %>%
  mutate(cells_per_uL = (raw_counts*dilution)/volume_uL,
         HNA_per_uL = (HNA_counts*dilution)/volume_uL,
         HNA_percent = (HNA_counts)/raw_counts*100,
         LNA_per_uL = (LNA_counts*dilution)/volume_uL,
         LNA_percent = (LNA_counts)/raw_counts*100,
         Nuc_acid_ratio = HNA_per_uL/LNA_per_uL)

all_flow_cy$Season[all_flow_cy$Season == "Winter"] = "Fall"
all_flow_cy$Month[all_flow_cy$Month == "Januari"] = "October"

all_flow_cy$Site <- factor(all_flow_cy$Site,  levels = c("MM110","MM45","MM15","MOT", "MDP", "MBR", "MLB","MIN"))

###  HNA PER uL
lim_all_HNA_per_uL <- c(min(all_flow_cy$HNA_per_uL) - 100,max(all_flow_cy$HNA_per_uL) + 100)


# Set up the main plot
all_Lakes_surface_HNA <- ggplot(filter(all_flow_cy, Depth == "Surface"), 
       aes(x = Site, y = HNA_per_uL, fill = Site, color = Season)) + 
  geom_boxplot(alpha = 0.5, color = "black", show.legend = FALSE) + 
  scale_y_continuous(limits = lim_all_HNA_per_uL) + 
  scale_fill_viridis(option = "viridis", discrete = TRUE) +
  scale_color_brewer(palette="Set1") +
  geom_jitter(size = 3.5) + facet_grid(.~Year, scales = "free_x") + ggtitle("Surface samples only")

all_Lakes_deep_HNA <- ggplot(filter(all_flow_cy, Depth == "Deep"), 
       aes(x = Site, y = HNA_per_uL, fill = Site, color = Season)) + 
  geom_boxplot(alpha = 0.5, color = "black", show.legend = FALSE) + 
  scale_y_continuous(limits = lim_all_HNA_per_uL) + 
  scale_fill_viridis(option = "viridis", discrete = TRUE) +
  scale_color_brewer(palette="Set1") +
  geom_jitter(size = 3.5) + facet_grid(.~Year, scales = "free_x") + ggtitle("Deep samples only")

all_Lakes_all_HNA <- ggplot(filter(all_flow_cy), 
       aes(x = Site, y = HNA_per_uL, fill = Site, color = Season)) + 
  geom_boxplot(alpha = 0.5, color = "black", show.legend = FALSE) +
  scale_y_continuous(limits = lim_all_HNA_per_uL) + 
  scale_fill_viridis(option = "viridis", discrete = TRUE) +
  scale_color_brewer(palette="Set1") +
  geom_jitter(size = 3.5) + facet_grid(.~Year, scales = "free_x") + ggtitle("All samples")


all_Lakes_HNA_per_uL_plots <- plot_grid(all_Lakes_surface_HNA, all_Lakes_deep_HNA, all_Lakes_all_HNA,
          labels = c("A", "B", "C"), 
          align = "h", nrow = 3, ncol = 1)


# HNA: LNA RATIO
lim_all_HNA <- c(min(all_flow_cy$HNA_percent) - 0.05,max(all_flow_cy$HNA_percent) + 0.05)

all_Lakes_surface_HNA_percent <- ggplot(filter(all_flow_cy, Depth == "Surface"), 
       aes(x = Site, y = HNA_percent, fill = Site, color = Season)) + 
  geom_boxplot(alpha = 0.5, color = "black") + 
  scale_y_continuous(limits = lim_all_HNA) + 
  scale_fill_viridis(option = "viridis", discrete = TRUE) +
  scale_color_brewer(palette="Set1") +
  geom_jitter(size = 3.5) + facet_grid(.~Year, scales = "free_x") + ggtitle("Surface samples only")

all_Lakes_deep_HNA_percent <- ggplot(filter(all_flow_cy, Depth == "Deep"), 
       aes(x = Site, y = HNA_percent, fill = Site, color = Season)) + 
  geom_boxplot(alpha = 0.5, color = "black") + 
  scale_y_continuous(limits = lim_all_HNA) + 
  scale_fill_viridis(option = "viridis", discrete = TRUE) +
  scale_color_brewer(palette="Set1") +
  geom_jitter(size = 3.5) + facet_grid(.~Year, scales = "free_x") + ggtitle("Deep samples only")

all_Lakes_all_HNA_percent <- ggplot(filter(all_flow_cy), 
       aes(x = Site, y = HNA_percent, fill = Site, color = Season)) + 
  geom_boxplot(alpha = 0.5, color = "black") +
  scale_y_continuous(limits = lim_all_HNA) + 
  scale_fill_viridis(option = "viridis", discrete = TRUE) +
  scale_color_brewer(palette="Set1") +
  geom_jitter(size = 3.5) + facet_grid(.~Year, scales = "free_x") + ggtitle("All samples")


all_Lakes_HNA_percent_plots <- plot_grid(all_Lakes_surface_HNA_percent,all_Lakes_deep_HNA_percent, all_Lakes_all_HNA_percent,
          labels = c("A", "B", "C"), 
          align = "h", nrow = 3, ncol = 1)



# DO EVERYTHING 
ALLTHEPLOTS <- plot_grid(all_Lakes_HNA_per_uL_plots,all_Lakes_HNA_percent_plots, 
          labels = c("1", "2"), 
          align = "h", nrow = 1, ncol = 2)
ALLTHEPLOTS


#  IS there a relationship between total counts and HNA?
a <- ggplot(all_flow_cy, aes(x = cells_per_uL, y = HNA_per_uL, fill = Site, color = Site, shape = Season)) + 
  geom_point(size = 3.5) + ggtitle("HNA per uL") + 
  scale_color_viridis(option = "viridis", discrete = TRUE)

b <- ggplot(all_flow_cy, aes(x = cells_per_uL, y = HNA_percent, fill = Site, color = Site, shape = Season)) + 
  geom_point(size = 3.5) + ggtitle("Percent HNA")+ 
  scale_color_viridis(option = "viridis", discrete = TRUE)

c <- ggplot(all_flow_cy, aes(x = cells_per_uL, y = LNA_per_uL, fill = Site, color = Site, shape = Season)) + 
  geom_point(size = 3.5) + ggtitle("LNA per uL") + 
  scale_color_viridis(option = "viridis", discrete = TRUE)

d <- ggplot(all_flow_cy, aes(x = cells_per_uL, y = LNA_percent, fill = Site, color = Site, shape = Season)) + 
  geom_point(size = 3.5) + ggtitle("Percent LNA")+ 
  scale_color_viridis(option = "viridis", discrete = TRUE)

e <- plot_grid(a,b,c, d, 
          labels = c("A", "B", "C", "D"), 
          align = "h", nrow = 2, ncol = 2)
e


new <- gather(all_flow_cy, "Nucleic_Acid_Type", "NA_cells_per_uL", LNA_per_uL, HNA_per_uL) %>%
  select(NA_cells_per_uL, Nucleic_Acid_Type, cells_per_uL, Month, Site, Season,Depth,Year)



p1 <- ggplot(new, aes(x = cells_per_uL, y = NA_cells_per_uL, color = Nucleic_Acid_Type)) + 
  geom_point(size = 3, alpha = 0.8) + 
  scale_color_manual(values = c("#675B78", "#EB7B88")) + 
  theme(legend.position = c(0.3, 0.85))

p2 <- ggplot(new, aes(x = cells_per_uL, y = NA_cells_per_uL, color = Season)) + 
  geom_point(size = 3, alpha = 0.8) + 
  scale_color_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) + 
  theme(legend.position = c(0.15, 0.85))


# Ratio of HNA:LNA is not dependent on the number cells per uL!!!!
p3 <- ggplot(all_flow_cy, aes(x = cells_per_uL, y = Nuc_acid_ratio, color = Season)) + 
  geom_point(size = 3, alpha = 0.8) + 
  scale_color_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  geom_smooth(method = "lm") +
  theme(legend.position = c(0.15, 0.85))

p4 <- ggplot(all_flow_cy, aes(x = cells_per_uL, y = Nuc_acid_ratio, color = Season, shape = as.factor(Year))) + 
  geom_point(size = 3, alpha = 0.8) + 
  scale_color_manual(values = c("#51BAAE", "#241B50", "#FD0F27")) +
  geom_smooth(method = "lm") +
  theme(legend.position = c(0.85, 0.75))


ggplot(all_flow_cy, aes(x = cells_per_uL, y = Nuc_acid_ratio, color = as.factor(Year), shape = Season)) + 
  geom_point(size = 5, alpha = 0.8) + 
  scale_color_manual(values = c("#FD7400","#211426", "#BEDB39")) + 
  theme(legend.position = c(0.15, 0.85))


p5 <- plot_grid(p1, p2, p3, p4, 
          labels = c("A", "B", "C", "D"), 
          align = "h", nrow = 2, ncol = 2)

#ggsave("Figures/HNA_LNA.png", p5, dpi = 600, units = "in", width = 10, height = 8)
```





